<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.56">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Kwangmin Kim">
<meta name="dcterms.date" content="2022-12-20">
<meta name="description" content="I could not share real data due to security issues, so I made simulation data to show a rough analysis methodology used for this projects. Therefore, this simulation data simply reflects generally known facts and the ones published on wikipedia, but does not reflect the distribution of sample data that was actually used, so the information derived from this data may differ from generally known biological or biomedical facts. 보안 문제로 real data를 공유하진 못하여 대략적인 분석 방법론을 보여주기위해 시뮬래이션 데이터를 만들었습니다. 따라서, 이 시뮬래이션 데이터는 일반적으로 알려진 사실과 wikipedia에 공개된 사실을 간단히 반영했지만 실제로 사용됐던 sample data의 분포를 반영하지 않았기 때문에 이 데이터에서 나온 정보는 일반적으로 알려진 생물학적 그리고 의학적 사실과 다를 수 있습니다.">

<title>Data Preparation – Kwangmin Kim</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script src="../../../site_libs/quarto-diagram/mermaid.min.js"></script>
<script src="../../../site_libs/quarto-diagram/mermaid-init.js"></script>
<link href="../../../site_libs/quarto-diagram/mermaid.css" rel="stylesheet">

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../.././images/logo.png" alt="" class="navbar-logo">
    </a>
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">Kwangmin Kim</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../docs/CV/index.html"> 
<span class="menu-text">CV</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../docs/projects/index.html"> 
<span class="menu-text">Projects</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../about.html"> 
<span class="menu-text">Me</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/kmink3225"> <i class="bi bi-github" role="img" aria-label="Github">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://www.linkedin.com/in/kwangmin-kim-a5241b200/"> <i class="bi bi-linkedin" role="img" aria-label="Linkedin">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#simulation-flowchart" id="toc-simulation-flowchart" class="nav-link active" data-scroll-target="#simulation-flowchart"><span class="header-section-number">1</span> Simulation Flowchart</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content column-page-left" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Data Preparation</h1>
<p class="subtitle lead">Simulation</p>
</div>

<div>
  <div class="description">
    I could not share real data due to security issues, so I made simulation data to show a rough analysis methodology used for this projects. Therefore, this simulation data simply reflects generally known facts and the ones published on wikipedia, but does not reflect the distribution of sample data that was actually used, so the information derived from this data may differ from generally known biological or biomedical facts. 보안 문제로 real data를 공유하진 못하여 대략적인 분석 방법론을 보여주기위해 시뮬래이션 데이터를 만들었습니다. 따라서, 이 시뮬래이션 데이터는 일반적으로 알려진 사실과 wikipedia에 공개된 사실을 간단히 반영했지만 실제로 사용됐던 sample data의 분포를 반영하지 않았기 때문에 이 데이터에서 나온 정보는 일반적으로 알려진 생물학적 그리고 의학적 사실과 다를 수 있습니다.
  </div>
</div>


<div class="quarto-title-meta column-page-left">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Kwangmin Kim </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">December 20, 2022</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="simulation-flowchart" class="level2" data-number="1">
<h2 data-number="1" class="anchored" data-anchor-id="simulation-flowchart"><span class="header-section-number">1</span> Simulation Flowchart</h2>
<div class="cell" data-fig-width="7.5" data-fig-height="9" data-layout-align="default">
<div class="cell-output-display">
<div>
<p></p><figure class="figure"><p></p>
<div>
<pre class="mermaid mermaid-js">flowchart TB
    subgraph Simulation
        direction TB
        subgraph Assign_Setting_Values
            direction LR
            Assign_Sample_Size---
            Assign_Dimension_Size---
            Assign_Covariance_Correlation---
            Assign_Several_Proportions---
            Assign_Noise_Intensity
        end
        subgraph Generate_Metabolite_Variables
            direction LR
            Generate_Covariance_Matrix---
            Apply_Noise_to_Covariance---
            Generate_Weights_Matrix---
            Use_MVN_Distribution---
            Generate_Metabolite_Data
        end
        subgraph Generate_Outcome_Variable
            direction LR
            Calculate_Score_Matrix---
            Use_Logit_Link---
            Calculate_Outcome_Probabilities---
            Use_Binomial_Distribution1---
            Generate_Binary_Outcome_Data
        end
        subgraph Generate_Sex_Variable
            direction LR
            Use_Binomial_Distribution2---
            Generate_Sex_Data
        end
        subgraph Generate_Age_Variable
            direction LR
            Search_the_Strongest_Metabolite---
            Rescale_It_to_Age---
            Generate_Age_Data
        end
        subgraph Generate_Genotype_Variable
            direction LR       
            Calculate_Marginal_Proportions---
            Calcualte_Joint_Proportions---
            Generate_Genotype_data
        end
        subgraph Merge_All_Data
            direction LR
            Outcome_Variable---
            Sex_Variable---
            Age_Variable---
            Genotype_Variable---
            Metabolite_Data
        end
        Assign_Setting_Values--&gt;Generate_Metabolite_Variables--&gt;Generate_Outcome_Variable--&gt;
        Generate_Sex_Variable--&gt;Generate_Age_Variable--&gt;Generate_Genotype_Variable--&gt;
        Merge_All_Data
    end
    subgraph Data_Analytics
        direction LR
        exploratory_data_analysis---
        statistical_analysis---
        machine_learning
    end
    subgraph Conclusion
        direction LR
    end
    Simulation--&gt;Data_Analytics--&gt;Conclusion
</pre>
</div>
<p></p></figure><p></p>
</div>
</div>
</div>
<ul>
<li>MVN: Multivariate Normal Distirubtion</li>
</ul>
<ul class="nav nav-pills" id="language-tab" role="tablist">
<li class="nav-item" role="presentation">
<button class="nav-link active" id="Korean-tab" data-bs-toggle="tab" data-bs-target="#Korean" type="button" role="tab" aria-controls="Korean" aria-selected="true">
<p>Korean</p>
</button>
</li>
<li class="nav-item" role="presentation">
<button class="nav-link" id="English-tab" data-bs-toggle="tab" data-bs-target="#English" type="button" role="tab" aria-controls="knitr" aria-selected="false">
<p>English</p>
</button>
</li>
<div id="language-tabcontent" class="tab-content">
<div id="Korean" class="tab-pane fade show active" role="tabpanel" aria-labelledby="Korean-tab">
<p>대략적인 분석 방법론을 간단히 보여주기 위해 Simulation을 수행했다. 이해를 돕기위해 Simulation 순서도를 간략히 설명하자면 크게 7 단계로 Simulation을 수행했다.</p>
<section id="설정값-생성" class="level4" data-number="1.0.1">
<h4 data-number="1.0.1" class="anchored" data-anchor-id="설정값-생성"><span class="header-section-number">1.0.1</span> 설정값 생성</h4>
<p><strong>Data Set Size Setting</strong></p>
<p>Simulation에 필요한 몇 가지 설정값들을 Global Variables로 설정하여 후차적으로 작성된 스크립트에서 호출이 자유롭도록 작성했다. 변수들은 아래의 Simulation section에 있는 <strong>Global Variables</strong> (see <a href="#sec-global_variables" class="quarto-xref">Section&nbsp;2.2</a>) 에서 확인 가능하다.</p>
<p><strong>Categorial Data Setting</strong></p>
<p>먼저, 고차원 데이터의 차원을 설정하기 위해 Sample Size와 변수의 수를 설정한 후 Categorical predictors를 만들기 위해 잘 알려진 분포, 내가 정한 분포, 혹은 임의로 발생하게 만든 분포를 설정하였다. (see <a href="#sec-global_variables" class="quarto-xref">Section&nbsp;2.2</a>, <a href="#sec-functions" class="quarto-xref">Section&nbsp;2.3</a>, and <a href="#sec-simulation" class="quarto-xref">Section&nbsp;2.4</a>)</p>
<p><strong>Sex Variable Setting</strong></p>
<p>Sex 변수는 <span class="math inline">\(X \sim \text{Bernoulli}(0.5)\)</span> 을 통해 data를 생성했다. (see <a href="#sec-simulation" class="quarto-xref">Section&nbsp;2.4</a>)</p>
<p><strong>Genotype Variable Setting</strong></p>
<p>Genotype 변수의 data는 아직도 어떻게 통계적으로 생성해야하는지 감을 못잡은 상태이기 때문에 더 연구가 필요하다. 하지만, 질병에 대한 유전적 영향도는 반영해야하기 때문에 outcome variable과 이미 잘 알려진 genotype의 분포를 반영하려고 노력했다. 두 변수에 연관성을 갖게하기 위해 각 변수의 proportion을 marginal distirubtion으로 설정하여 두 변수의 joint proportion을 계산하여 Genotype data를 생성했다. (see <a href="#sec-functions" class="quarto-xref">Section&nbsp;2.3</a> and <a href="#sec-simulation" class="quarto-xref">Section&nbsp;2.4</a>)</p>
<p><strong>Metabolite Data Setting</strong></p>
<p>고차원의 metabolite data를 만드는 설정으로, 고차원이면서 그룹내 서로 상관 관계가 있는 변수들을 생성하기 위해 난수에 의해 발생되는 임의의 Covariance를 생성하여 MVN (Multivariate Normal Distribution)에 반영되게 했고 각 그룹의 반응 변수로의 영향(또는 가중치)도 또한 난수로 임의적으로 발생되게 설정했다. 이때, 난수에 의해 임의적으로 발생하는 수치는 내가 임의적으로 범위를 한정했다. 재현성을 위해 seed number를 고정했다. (see <a href="#sec-simulation" class="quarto-xref">Section&nbsp;2.4</a>)</p>
<p><strong>Outcome Variable Setting</strong></p>
<p>MVN에 의해 만들어진 Data와 미리 만들어 놓은 가중치 Matrix의 곱을 통해 Score Matrix를 만들고 Logit Link를 이용하여 각 Sample의 확률값을 만들었다. 각 Sample의 확률값을 기반으로 <span class="math inline">\(X \sim \text{Bernoulli}(p)\)</span>, (여기서 <span class="math inline">\(p\)</span>는 각 sample이 갖는 확률값을 뜻한다), 을 통해 disease status의 정보를 담은 binary outcome variable를 만들었다. (see <a href="#sec-simulation" class="quarto-xref">Section&nbsp;2.4</a>)</p>
<p><strong>Age Variable Setting</strong></p>
<p>Age 변수는 생물학적, 의학적으로 치매와 연관성이 높은 요인으로 Outcome 변수로 가장 설명이 잘되는 metabolite를 탐색해 선별하여 Age 형태로 변환을 했다. 제일 어린 사람을 65세 그리고 제일 연장자를 105세로 설정하여 min max normalization을 적용했다. (see (see <a href="#sec-functions" class="quarto-xref">Section&nbsp;2.3</a> and <a href="#sec-simulation" class="quarto-xref">Section&nbsp;2.4</a>)</p>
<p><strong>Merge All Data</strong></p>
<p>Simulation을 통해 만들어진 각 변수들을 index를 만들어 병합시켜 data frame의 형태로 만들었다. (see <a href="#sec-simulation" class="quarto-xref">Section&nbsp;2.4</a>)</p>
<p><strong>Analytics &amp; Conclusion</strong></p>
<p>분석 부분은 이 data preparation section에서는 자세히 기술하지 않고 EDA, Statistical Approach 및 ML Approach Section에서 자세히 다룰예정이다. 간략히 말하면, outcome 변수와 통계적으로 유의한 관계를 갖는 metabolite를 선별하고 그 결과가 machine learning을 이용하여 얻은 결과와 얼마나 같은지 비교 분석을 하여 Outcome variable에 가장 연관성이 있는 변수들을 규명하는 방법을 기술할 예정이다.</p>
</section>
<section id="simulation" class="level2" data-number="2">
<h2 data-number="2" class="anchored" data-anchor-id="simulation"><span class="header-section-number">2</span> Simulation</h2>
<section id="package-loading-and-option-settings" class="level3" data-number="2.1">
<h3 data-number="2.1" class="anchored" data-anchor-id="package-loading-and-option-settings"><span class="header-section-number">2.1</span> Package Loading and Option Settings</h3>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(janitor)) <span class="fu">install.packages</span>(<span class="st">"janitor"</span>) </span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(tidyverse)) <span class="fu">install.packages</span>(<span class="st">"tidyverse"</span>) </span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(tidymodels)) <span class="fu">install.packages</span>(<span class="st">"tidymodels"</span>) </span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(glmnet)) <span class="fu">install.packages</span>(<span class="st">"glmnet"</span>) </span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(MASS)) <span class="fu">install.packages</span>(<span class="st">"MASS"</span>) </span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(ggpubr)) <span class="fu">install.packages</span>(<span class="st">"ggpubr"</span>) </span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(car)) <span class="fu">install.packages</span>(<span class="st">"car"</span>) </span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(mixOmics)) <span class="fu">install.packages</span>(<span class="st">"mixOmics"</span>) </span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(janitor)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidymodels)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmnet)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MASS)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr) </span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(car) </span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mixOmics)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">20230103</span>) </span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">message=</span><span class="cn">FALSE</span>,<span class="at">warning=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="sec-global_variables" class="level3" data-number="2.2">
<h3 data-number="2.2" class="anchored" data-anchor-id="sec-global_variables"><span class="header-section-number">2.2</span> Global Variables</h3>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># the number of samples</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>sample_size <span class="ot">&lt;-</span> <span class="dv">500</span> <span class="co">#1000</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co"># the number of predictors</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>predictor_size <span class="ot">&lt;-</span> <span class="dv">1000</span> <span class="co">#5000</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co"># the number of groups</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>group_size <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="dv">6</span><span class="sc">:</span><span class="dv">10</span>,<span class="dv">1</span>) <span class="co"># at least more than 6, the number of the genotypes</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="co"># the number of predictors truly associated with a response variable</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>significant_predictors <span class="ot">&lt;-</span> <span class="fu">floor</span>(predictor_size<span class="sc">*</span><span class="fu">sample</span>((<span class="dv">50</span><span class="sc">:</span><span class="dv">100</span>)<span class="sc">/</span><span class="dv">1000</span>,<span class="dv">1</span>)) </span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="do">## set the predictors associated with an outcome</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="do">### the number of predictors positively associated with an outcome</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="do">### the number of predictors negatively associated with an outcome</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>positively_associated_predictors<span class="ot">&lt;-</span><span class="fu">floor</span>(significant_predictors<span class="sc">*</span><span class="fl">0.4</span>) </span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>negatively_associated_predictors<span class="ot">&lt;-</span>significant_predictors<span class="sc">-</span>positively_associated_predictors </span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a><span class="do">## set the proportion of the groups in which the predictors are correlated with one another</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a><span class="do">### randomly sampling proportions to become their sum equal to 1</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>group_proportion_list<span class="ot">&lt;-</span><span class="fu">sample</span>(<span class="fu">seq</span>(<span class="dv">1</span>,<span class="dv">1</span><span class="sc">+</span><span class="dv">2</span><span class="sc">*</span>(<span class="dv">100</span><span class="sc">-</span>group_size)<span class="sc">/</span>group_size,</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>                            <span class="at">by=</span><span class="dv">2</span><span class="sc">*</span>(<span class="dv">100</span><span class="sc">-</span>group_size)<span class="sc">/</span>(group_size<span class="sc">*</span>(group_size<span class="dv">-1</span>)))<span class="sc">/</span><span class="dv">100</span>,</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>                        group_size,<span class="at">replace=</span><span class="cn">FALSE</span>)<span class="sc">%&gt;%</span><span class="fu">round</span>(<span class="dv">3</span>) </span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(group_proportion_list)<span class="ot">&lt;-</span><span class="fu">paste0</span>(<span class="st">"group"</span>,<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(group_proportion_list))</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a><span class="do">### initialize a matrix with a size as sample_size by predictor_size</span></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>predictor_matrix <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">0</span>, <span class="at">ncol =</span> predictor_size, <span class="at">nrow =</span> sample_size)</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a><span class="do">### initialize a data frame and assign meta information used to generate simulated data</span></span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>group_meta_data<span class="ot">&lt;-</span></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(</span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">group_name=</span><span class="fu">c</span>(<span class="fu">names</span>(group_proportion_list)),</span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>        <span class="at">proportion=</span>group_proportion_list)<span class="sc">%&gt;%</span></span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mutate</span>(</span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>            <span class="co"># the number of predictors within each group </span></span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>            <span class="at">group_n=</span>(predictor_size<span class="sc">*</span>group_proportion_list)<span class="sc">%&gt;%</span><span class="fu">round</span>(<span class="dv">0</span>),</span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>            <span class="co"># the 1st index of predictors in each group</span></span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>            <span class="at">first_index=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="fu">cumsum</span>(group_n[<span class="sc">-</span><span class="fu">length</span>(group_proportion_list)])<span class="sc">+</span><span class="dv">1</span>), </span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>            <span class="co"># the last index of predictors in each group</span></span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>            <span class="at">last_index=</span><span class="fu">cumsum</span>(group_n),</span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>            <span class="co"># within-group correlations among the within-group predictors </span></span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a>            <span class="at">group_correlation=</span><span class="fu">sample</span>((<span class="dv">0</span><span class="sc">:</span><span class="dv">700</span>)<span class="sc">/</span><span class="dv">1000</span>,<span class="fu">length</span>(group_proportion_list),<span class="at">replace=</span><span class="cn">TRUE</span>),</span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>            <span class="co"># effect of each group on an outcome variable </span></span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a>            <span class="at">group_effect=</span><span class="fu">sample</span>((<span class="sc">-</span><span class="dv">40</span><span class="sc">:</span><span class="dv">30</span>)<span class="sc">/</span><span class="dv">100</span>,<span class="fu">length</span>(group_proportion_list),<span class="at">replace=</span><span class="cn">TRUE</span>)) </span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a><span class="do">### set a group effect as 0.7 into a group with the smallest group number </span></span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a>group_meta_data[<span class="fu">which.min</span>(group_meta_data[,<span class="st">"group_n"</span>]),<span class="st">"group_effect"</span>]<span class="ot">&lt;-</span><span class="fl">0.7</span></span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a><span class="do">### set a group effect as -0.5 into a group with the second smallest group number </span></span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a>group_meta_data[group_meta_data[,<span class="st">"group_n"</span>]<span class="sc">==</span>(<span class="fu">sort</span>(group_meta_data[,<span class="st">"group_n"</span>])[<span class="dv">2</span>]),<span class="st">"group_effect"</span>]<span class="ot">&lt;-</span>(<span class="sc">-</span><span class="fl">0.5</span>)</span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a><span class="co"># initialize a data matrix to assign simulated values</span></span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a><span class="do">## add some noise to data</span></span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a>data<span class="ot">&lt;-</span><span class="fu">matrix</span>(<span class="fu">rnorm</span>(sample_size<span class="sc">*</span>predictor_size,<span class="at">mean=</span><span class="dv">0</span>,<span class="at">sd=</span><span class="fl">0.05</span>), </span>
<span id="cb2-49"><a href="#cb2-49" aria-hidden="true" tabindex="-1"></a>             <span class="at">nrow =</span> sample_size, <span class="at">ncol =</span> predictor_size)</span>
<span id="cb2-50"><a href="#cb2-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-51"><a href="#cb2-51" aria-hidden="true" tabindex="-1"></a><span class="co"># initialize a covariance matrix to assign simulated values</span></span>
<span id="cb2-52"><a href="#cb2-52" aria-hidden="true" tabindex="-1"></a>covariance_matrix<span class="ot">&lt;-</span><span class="fu">matrix</span>(<span class="fu">rnorm</span>(predictor_size<span class="sc">*</span>predictor_size,<span class="at">mean=</span><span class="dv">0</span>,<span class="at">sd=</span><span class="fl">0.05</span>),</span>
<span id="cb2-53"><a href="#cb2-53" aria-hidden="true" tabindex="-1"></a>                          <span class="at">nrow=</span>predictor_size, <span class="at">ncol=</span>predictor_size)</span>
<span id="cb2-54"><a href="#cb2-54" aria-hidden="true" tabindex="-1"></a>beta_coefficients <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(predictor_size,<span class="dv">0</span>,<span class="fl">0.05</span>)</span>
<span id="cb2-55"><a href="#cb2-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-56"><a href="#cb2-56" aria-hidden="true" tabindex="-1"></a>answer_list<span class="ot">&lt;-</span><span class="fu">list</span>(</span>
<span id="cb2-57"><a href="#cb2-57" aria-hidden="true" tabindex="-1"></a>    <span class="st">'sample size'</span><span class="ot">=</span>sample_size,</span>
<span id="cb2-58"><a href="#cb2-58" aria-hidden="true" tabindex="-1"></a>    <span class="st">'predictor size'</span><span class="ot">=</span>predictor_size,</span>
<span id="cb2-59"><a href="#cb2-59" aria-hidden="true" tabindex="-1"></a>    <span class="st">'group size'</span><span class="ot">=</span>group_size,</span>
<span id="cb2-60"><a href="#cb2-60" aria-hidden="true" tabindex="-1"></a>    <span class="st">'significant predictors'</span><span class="ot">=</span>significant_predictors,</span>
<span id="cb2-61"><a href="#cb2-61" aria-hidden="true" tabindex="-1"></a>    <span class="st">'positively associated predictors'</span><span class="ot">=</span>positively_associated_predictors,</span>
<span id="cb2-62"><a href="#cb2-62" aria-hidden="true" tabindex="-1"></a>    <span class="st">'negatively associated predictors'</span><span class="ot">=</span>negatively_associated_predictors,</span>
<span id="cb2-63"><a href="#cb2-63" aria-hidden="true" tabindex="-1"></a>    <span class="st">'group proportion list'</span><span class="ot">=</span>group_proportion_list,</span>
<span id="cb2-64"><a href="#cb2-64" aria-hidden="true" tabindex="-1"></a>    <span class="st">'group meta data'</span><span class="ot">=</span>group_meta_data,</span>
<span id="cb2-65"><a href="#cb2-65" aria-hidden="true" tabindex="-1"></a>    <span class="st">'data noise intensity'</span><span class="ot">=</span><span class="fu">c</span>(<span class="st">'distribution'</span><span class="ot">=</span><span class="st">'rnorm'</span>,<span class="st">'mean'</span><span class="ot">=</span><span class="dv">0</span>,<span class="st">'sd'</span><span class="ot">=</span><span class="fl">0.05</span>),</span>
<span id="cb2-66"><a href="#cb2-66" aria-hidden="true" tabindex="-1"></a>    <span class="st">'covariance noise intensity'</span><span class="ot">=</span><span class="fu">c</span>(<span class="st">'distribution'</span><span class="ot">=</span><span class="st">'rnorm'</span>,<span class="st">'mean'</span><span class="ot">=</span><span class="dv">0</span>,<span class="st">'sd'</span><span class="ot">=</span><span class="fl">0.05</span>),</span>
<span id="cb2-67"><a href="#cb2-67" aria-hidden="true" tabindex="-1"></a>    <span class="st">'effect noise intensity on response'</span><span class="ot">=</span><span class="fu">c</span>(<span class="st">'distribution'</span><span class="ot">=</span><span class="st">'rnorm'</span>,<span class="st">'mean'</span><span class="ot">=</span><span class="dv">0</span>,<span class="st">'sd'</span><span class="ot">=</span><span class="fl">0.05</span>),</span>
<span id="cb2-68"><a href="#cb2-68" aria-hidden="true" tabindex="-1"></a>    <span class="st">'link function between the response and predictors'</span> <span class="ot">=</span> <span class="st">'canonical logit link function'</span>,</span>
<span id="cb2-69"><a href="#cb2-69" aria-hidden="true" tabindex="-1"></a>    <span class="st">'link function noise intensity'</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'distribution'</span><span class="ot">=</span><span class="st">'rnorm'</span>,<span class="st">'mean'</span><span class="ot">=</span><span class="dv">0</span>,<span class="st">'sd'</span><span class="ot">=</span><span class="fl">0.05</span>),</span>
<span id="cb2-70"><a href="#cb2-70" aria-hidden="true" tabindex="-1"></a>    <span class="st">'age_distirbution'</span><span class="ot">=</span><span class="st">'used data of a variable with the highest effect on outcome'</span>,</span>
<span id="cb2-71"><a href="#cb2-71" aria-hidden="true" tabindex="-1"></a>    <span class="st">'sex_distribution'</span><span class="ot">=</span><span class="st">'rbinom(n=sample size,p=0.5)'</span>,</span>
<span id="cb2-72"><a href="#cb2-72" aria-hidden="true" tabindex="-1"></a>    <span class="st">'genotype_distirbution'</span><span class="ot">=</span><span class="fu">c</span>(<span class="st">'e2'</span> <span class="ot">=</span> <span class="st">'8.4%'</span>,<span class="st">'e3'</span> <span class="ot">=</span> <span class="st">'77.9%'</span>,<span class="st">'e4'</span> <span class="ot">=</span> <span class="st">'13.7%'</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="sec-functions" class="level3" data-number="2.3">
<h3 data-number="2.3" class="anchored" data-anchor-id="sec-functions"><span class="header-section-number">2.3</span> Functions</h3>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Function List</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>scale_function<span class="ot">=</span><span class="cf">function</span>(<span class="at">vector=</span>x,<span class="at">min=</span><span class="cn">NULL</span>,<span class="at">max=</span><span class="cn">NULL</span>,<span class="at">method=</span><span class="st">"customized"</span>){</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(method<span class="sc">==</span><span class="st">"min-max"</span>){</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>        result<span class="ot">=</span>(vector<span class="sc">-</span><span class="fu">min</span>(vector))(<span class="fu">max</span>(vector)<span class="sc">-</span><span class="fu">min</span>(vector))</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>    }<span class="cf">else</span> <span class="cf">if</span>(method<span class="sc">==</span><span class="st">"customized"</span>){</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>        result<span class="ot">=</span>(max<span class="sc">-</span>min)<span class="sc">*</span>(vector<span class="sc">-</span><span class="fu">min</span>(vector))<span class="sc">/</span>(<span class="fu">max</span>(vector)<span class="sc">-</span><span class="fu">min</span>(vector))<span class="sc">+</span>min</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>    }<span class="cf">else</span>{</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>        result<span class="ot">=</span>(vector<span class="sc">-</span><span class="fu">mean</span>(vector))<span class="sc">/</span><span class="fu">sd</span>(vector)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(result)</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>age_data_generator<span class="ot">=</span><span class="cf">function</span>(in_data,in_response,<span class="at">fun=</span>scale_function){</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>    <span class="co"># this function generates a age (continuous) data </span></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>    <span class="co"># that are statistically associated with a simulated variable as designed above.</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>    <span class="do">## conduct t test with the response and each variable generated by multivariate normal distributions.</span></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>    <span class="do">## search a variable with the largest difference in mean between the two groups or with the lowest p value</span></span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>    <span class="do">## In this case, I will pick the former one. </span></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>    <span class="do">## (I don't care about the multiple testing problems for now)</span></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>    temp_df<span class="ot">=</span><span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="at">ncol=</span><span class="dv">5</span>)) <span class="co"># initialize an empty data frame</span></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(in_data)){</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>        temp_df[i,]<span class="ot">=</span><span class="fu">c</span>(</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>            <span class="fu">names</span>(in_data)[i],</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>            <span class="fu">t.test</span>(in_data[,i]<span class="sc">~</span>in_response)<span class="sc">$</span>estimate[<span class="dv">1</span>],</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>            <span class="fu">t.test</span>(in_data[,i]<span class="sc">~</span>in_response)<span class="sc">$</span>estimate[<span class="dv">2</span>],</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>            <span class="fu">t.test</span>(in_data[,i]<span class="sc">~</span>in_response)<span class="sc">$</span>estimate[<span class="dv">2</span>]<span class="sc">-</span><span class="fu">t.test</span>(data[,i]<span class="sc">~</span>response)<span class="sc">$</span>estimate[<span class="dv">1</span>],</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>            <span class="fu">t.test</span>(in_data[,i]<span class="sc">~</span>in_response)<span class="sc">$</span>p.value)</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(temp_df)<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">'metabolite'</span>,<span class="st">'mean_neg'</span>,<span class="st">'mean_pos'</span>,<span class="st">'mean_diff'</span>,<span class="st">'p.value'</span>)</span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>    <span class="do">## search a variable with the largest difference in mean</span></span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>    strong_metabolite<span class="ot">&lt;-</span></span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a>        temp_df<span class="sc">%&gt;%</span></span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mutate</span>(</span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a>            <span class="at">mean_neg=</span><span class="fu">as.numeric</span>(mean_neg),</span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>            <span class="at">mean_pos=</span><span class="fu">as.numeric</span>(mean_pos),</span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a>            <span class="at">mean_diff=</span><span class="fu">as.numeric</span>(mean_diff),</span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>            <span class="at">p.value=</span><span class="fu">as.numeric</span>(p.value),</span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>            <span class="at">abs_mean_diff=</span><span class="fu">abs</span>(mean_diff))<span class="sc">%&gt;%</span></span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>        <span class="fu">filter</span>(abs_mean_diff<span class="sc">==</span><span class="fu">max</span>(abs_mean_diff))<span class="sc">%&gt;%</span></span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(metabolite)<span class="sc">%&gt;%</span>pull</span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>    <span class="do">## generate age data with min max normalization</span></span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a>    age_data<span class="ot">&lt;-</span></span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a>        data<span class="sc">%&gt;%</span></span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(strong_metabolite)<span class="sc">%&gt;%</span></span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a>        <span class="fu">scale_function</span>(<span class="at">vector=</span>.,<span class="at">min=</span><span class="dv">65</span>,<span class="at">max=</span><span class="dv">105</span>,<span class="at">method=</span><span class="st">"customized"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a>        <span class="fu">rename</span>(<span class="at">age=</span><span class="dv">1</span>)<span class="sc">%&gt;%</span><span class="fu">round</span>(<span class="dv">0</span>)</span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(age_data)</span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a>genotype_data_generator<span class="ot">=</span><span class="cf">function</span>(<span class="at">in_response=</span>response,<span class="at">fun=</span>scale_function){</span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a>    <span class="co"># this function generates a genotype (categorical) data </span></span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a>    <span class="co"># that are jointly and statistically associated with a continuous data and a binary data </span></span>
<span id="cb3-58"><a href="#cb3-58" aria-hidden="true" tabindex="-1"></a>    <span class="co"># (I am not so sure if I can generate data that are statistically associated with </span></span>
<span id="cb3-59"><a href="#cb3-59" aria-hidden="true" tabindex="-1"></a>    <span class="co"># some fake metabolite data. But, I will give it a try).</span></span>
<span id="cb3-60"><a href="#cb3-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-61"><a href="#cb3-61" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Declare the marginal proportions </span></span>
<span id="cb3-62"><a href="#cb3-62" aria-hidden="true" tabindex="-1"></a>    <span class="do">## for binary (affected vs unaffected) and a genotype (categorical) data, respectively</span></span>
<span id="cb3-63"><a href="#cb3-63" aria-hidden="true" tabindex="-1"></a>    binary_proportion<span class="ot">&lt;-</span><span class="fu">as.numeric</span>(<span class="fu">table</span>(in_response)<span class="sc">/</span>sample_size) <span class="co">#the simulated proportion for the disease vs non-disease cases</span></span>
<span id="cb3-64"><a href="#cb3-64" aria-hidden="true" tabindex="-1"></a>    genotype_proportion<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="fl">0.084</span>,<span class="fl">0.779</span>,<span class="fl">0.137</span>) <span class="co"># the known proportion of APOE genotypes from Wiki</span></span>
<span id="cb3-65"><a href="#cb3-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-66"><a href="#cb3-66" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Declare the joint proportion predictor matrix</span></span>
<span id="cb3-67"><a href="#cb3-67" aria-hidden="true" tabindex="-1"></a>    joint_proportion<span class="ot">&lt;-</span><span class="fu">matrix</span>(</span>
<span id="cb3-68"><a href="#cb3-68" aria-hidden="true" tabindex="-1"></a>        <span class="fu">c</span>(binary_proportion[<span class="dv">1</span>]<span class="sc">*</span>genotype_proportion, <span class="co"># for the unaffected cases</span></span>
<span id="cb3-69"><a href="#cb3-69" aria-hidden="true" tabindex="-1"></a>        binary_proportion[<span class="dv">2</span>]<span class="sc">*</span>genotype_proportion),  <span class="co"># for the affected cases</span></span>
<span id="cb3-70"><a href="#cb3-70" aria-hidden="true" tabindex="-1"></a>        <span class="at">ncol=</span><span class="dv">2</span>, <span class="at">byrow=</span><span class="cn">FALSE</span>)</span>
<span id="cb3-71"><a href="#cb3-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-72"><a href="#cb3-72" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Generate the genotype (catogrical) data</span></span>
<span id="cb3-73"><a href="#cb3-73" aria-hidden="true" tabindex="-1"></a>    genotype_data <span class="ot">=</span> <span class="fu">numeric</span>(sample_size) <span class="co"># initialize a vector </span></span>
<span id="cb3-74"><a href="#cb3-74" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>sample_size) {</span>
<span id="cb3-75"><a href="#cb3-75" aria-hidden="true" tabindex="-1"></a>        genotype_data[i] <span class="ot">=</span> <span class="fu">sample</span>(</span>
<span id="cb3-76"><a href="#cb3-76" aria-hidden="true" tabindex="-1"></a>            <span class="fu">c</span>(<span class="st">'e2'</span>,<span class="st">'e3'</span>,<span class="st">'e4'</span>),</span>
<span id="cb3-77"><a href="#cb3-77" aria-hidden="true" tabindex="-1"></a>             <span class="dv">1</span>, </span>
<span id="cb3-78"><a href="#cb3-78" aria-hidden="true" tabindex="-1"></a>             <span class="at">prob=</span>joint_proportion[,<span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">'neg'</span>,in_response[i]),<span class="dv">1</span>,<span class="dv">2</span>)])</span>
<span id="cb3-79"><a href="#cb3-79" aria-hidden="true" tabindex="-1"></a>    }    </span>
<span id="cb3-80"><a href="#cb3-80" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(genotype_data)</span>
<span id="cb3-81"><a href="#cb3-81" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="sec-simulation" class="level3" data-number="2.4">
<h3 data-number="2.4" class="anchored" data-anchor-id="sec-simulation"><span class="header-section-number">2.4</span> Simulation</h3>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># generate simulation data using multivariate normal distribution</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(group_meta_data)) {</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>    group_range <span class="ot">&lt;-</span> group_meta_data[i, <span class="st">"first_index"</span>]<span class="sc">:</span>group_meta_data[i, <span class="st">"last_index"</span>]</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (j <span class="cf">in</span> group_range){</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span>(k <span class="cf">in</span> group_range){</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>        covariance_matrix[j, k] <span class="ot">&lt;-</span> group_meta_data[i, <span class="st">"group_correlation"</span>]</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>    <span class="co">#covariance_matrix[group_range, group_range]+group_meta_data[i, "group_correlation"]    </span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">diag</span>(covariance_matrix) <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>    data[, group_range] <span class="ot">&lt;-</span> </span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mvrnorm</span>(<span class="at">n =</span> sample_size, </span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>                <span class="at">mu =</span> <span class="fu">rep</span>(<span class="dv">0</span>,group_meta_data[i,<span class="st">"group_n"</span>]),</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>                <span class="at">Sigma =</span> covariance_matrix[group_range, group_range])</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>    data<span class="ot">=</span><span class="fu">as.data.frame</span>(data)</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>    beta_coefficients[group_meta_data[i, <span class="st">"first_index"</span>]<span class="sc">:</span>group_meta_data[i, <span class="st">"last_index"</span>]] <span class="ot">&lt;-</span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>        beta_coefficients[group_meta_data[i, <span class="st">"first_index"</span>]<span class="sc">:</span>group_meta_data[i, <span class="st">"last_index"</span>]]<span class="sc">+</span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>        group_meta_data[i,<span class="st">"group_effect"</span>]</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>    predictor_names<span class="ot">&lt;-</span><span class="fu">paste0</span>(group_meta_data[i,<span class="st">"group_name"</span>],<span class="st">"_"</span>,<span class="dv">1</span><span class="sc">:</span>group_meta_data[i,<span class="st">"group_n"</span>])</span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(beta_coefficients)[group_meta_data[i, <span class="st">"first_index"</span>]<span class="sc">:</span>group_meta_data[i, <span class="st">"last_index"</span>]] <span class="ot">&lt;-</span> predictor_names</span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(data)[group_meta_data[i, <span class="st">"first_index"</span>]<span class="sc">:</span>group_meta_data[i, <span class="st">"last_index"</span>]]<span class="ot">&lt;-</span>predictor_names</span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>score<span class="ot">=</span><span class="fu">as.matrix</span>(data)<span class="sc">%*%</span>beta_coefficients <span class="co"># score of each sample</span></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a><span class="co"># logistic function to get a probability, intercept = 0, </span></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a><span class="do">## set probabilities-0.2 to apply noise and negative probabilities into 0</span></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>probabilities <span class="ot">&lt;-</span> </span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>    ((<span class="dv">1</span><span class="sc">/</span>(<span class="dv">1</span><span class="sc">+</span><span class="fu">exp</span>(<span class="sc">-</span>(<span class="dv">0</span><span class="sc">+</span>score))))<span class="sc">-</span><span class="fu">rnorm</span>(sample_size,<span class="at">m=</span><span class="fl">0.2</span>,<span class="at">sd=</span><span class="fl">0.05</span>))<span class="sc">%&gt;%</span></span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ifelse</span>(.<span class="sc">&gt;</span><span class="dv">1</span>,<span class="dv">1</span>,.)<span class="sc">%&gt;%</span></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ifelse</span>(.<span class="sc">&lt;</span><span class="dv">0</span>,<span class="dv">0</span>,.)</span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a>response <span class="ot">&lt;-</span></span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ifelse</span>(<span class="fu">rbinom</span>(sample_size, <span class="dv">1</span>, probabilities)<span class="sc">==</span><span class="dv">0</span>,<span class="st">'negative'</span>,<span class="st">'positive'</span>)<span class="sc">%&gt;%</span></span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">factor</span>(.,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">'negative'</span>,<span class="st">'positive'</span>))</span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a><span class="co"># simulate sex data</span></span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a>sex_data <span class="ot">&lt;-</span> <span class="fu">rbinom</span>(sample_size,<span class="dv">1</span>,<span class="fl">0.5</span>)</span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a><span class="co"># simulate age data</span></span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a>age_data <span class="ot">&lt;-</span> <span class="fu">age_data_generator</span>(<span class="at">in_data=</span>data,<span class="at">in_response=</span>response)</span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a><span class="co"># simulate genotype data</span></span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a>genotype_data <span class="ot">&lt;-</span> <span class="fu">genotype_data_generator</span>(<span class="at">in_response=</span>response)</span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a><span class="co"># merge the univariables</span></span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a>phenotype_data<span class="ot">&lt;-</span></span>
<span id="cb4-49"><a href="#cb4-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(</span>
<span id="cb4-50"><a href="#cb4-50" aria-hidden="true" tabindex="-1"></a>        <span class="at">id=</span><span class="dv">1</span><span class="sc">:</span>sample_size,</span>
<span id="cb4-51"><a href="#cb4-51" aria-hidden="true" tabindex="-1"></a>        <span class="at">outcome=</span>response,</span>
<span id="cb4-52"><a href="#cb4-52" aria-hidden="true" tabindex="-1"></a>        <span class="at">age=</span>age_data,</span>
<span id="cb4-53"><a href="#cb4-53" aria-hidden="true" tabindex="-1"></a>        <span class="at">sex=</span>sex_data,</span>
<span id="cb4-54"><a href="#cb4-54" aria-hidden="true" tabindex="-1"></a>        <span class="at">genotype=</span>genotype_data)</span>
<span id="cb4-55"><a href="#cb4-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-56"><a href="#cb4-56" aria-hidden="true" tabindex="-1"></a>all_data<span class="ot">=</span><span class="fu">inner_join</span>(</span>
<span id="cb4-57"><a href="#cb4-57" aria-hidden="true" tabindex="-1"></a>    phenotype_data,</span>
<span id="cb4-58"><a href="#cb4-58" aria-hidden="true" tabindex="-1"></a>    data<span class="sc">%&gt;%</span><span class="fu">mutate</span>(<span class="at">id=</span><span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>()),</span>
<span id="cb4-59"><a href="#cb4-59" aria-hidden="true" tabindex="-1"></a>    <span class="at">by=</span><span class="st">"id"</span>)</span>
<span id="cb4-60"><a href="#cb4-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-61"><a href="#cb4-61" aria-hidden="true" tabindex="-1"></a><span class="co">#write_rds(all_data,"./docs/data/llfs_simulated_data.rds")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="load-data" class="level3" data-number="2.5">
<h3 data-number="2.5" class="anchored" data-anchor-id="load-data"><span class="header-section-number">2.5</span> Load Data</h3>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load simulation data</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>simulated_data<span class="ot">&lt;-</span><span class="fu">read_rds</span>(datapath)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co"># simple data pre-processing</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>all_data<span class="ot">&lt;-</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>    simulated_data<span class="sc">%&gt;%</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">outcome=</span><span class="fu">factor</span>(outcome,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">"negative"</span>,<span class="st">"positive"</span>)),</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">sex=</span><span class="fu">ifelse</span>(sex<span class="sc">==</span><span class="dv">0</span>,<span class="st">"man"</span>,<span class="st">"woman"</span>),</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>      <span class="at">sex=</span><span class="fu">factor</span>(sex,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">"man"</span>,<span class="st">"woman"</span>)),</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>      <span class="at">genotype=</span><span class="fu">factor</span>(genotype,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">"e3"</span>,<span class="st">"e2"</span>,<span class="st">"e4"</span>))</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="co"># rename metabolite variables</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(all_data)[<span class="dv">6</span><span class="sc">:</span><span class="fu">ncol</span>(all_data)]<span class="ot">&lt;-</span><span class="fu">paste0</span>(<span class="st">"meta"</span>,<span class="dv">1</span><span class="sc">:</span>predictor_size)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="data-description" class="level2" data-number="3">
<h2 data-number="3" class="anchored" data-anchor-id="data-description"><span class="header-section-number">3</span> Data Description</h2>
<p>This data include 500 samples and 1005 variables:</p>
<ul>
<li>id: sample ID.</li>
<li>outcome: a disease status (positive, negative), positive is an affected status, negative is an unaffected status, and the reference group is positive.</li>
<li>age: age</li>
<li>sex: sex (man, woman) and the reference group is man.</li>
<li>genotype: APOE genotypes
<ul>
<li>the apolipoprotein <span class="math inline">\(\epsilon\)</span> (APOE) is a protein produced in the metabolic pathway of fats in mammals, a genotype of which seems to be related to Alzheimer’s disease (AD). APOE is polymorphic and has three major alleles, <span class="math inline">\(\epsilon 2\)</span> (e2), <span class="math inline">\(\epsilon 3\)</span>(e3), and <span class="math inline">\(\epsilon 4\)</span> (e4). The statistics of the polymorphism are 8.4% for e2, 77.9% for e3, and 13.7% for e4 in worldwide allel frequency, respectively. It is known that the e2, e3, and e4 allels are associated with the protective factor, the neutral one, and the risk one with regard to AD. However, this finding has not been replicated in a large population. Therefore, it is known that we do not know their true associations with AD in the true population. <a href="https://en.wikipedia.org/wiki/Apolipoprotein_E">(from Wiki)</a>
<ul>
<li>There are 6 combinations of the genotypes:
<ul>
<li>e2/e2</li>
<li>e2/e3</li>
<li>e2/e4</li>
<li>e3/e3</li>
<li>e3/e4</li>
<li>e4/e4</li>
</ul></li>
<li>In this study, I generated the 3 major alleles, e3, e4, e2</li>
</ul></li>
</ul></li>
<li>meta1 ~ meta1000: a list of metabolites that were blood-sampled from the APOE carriers.</li>
</ul>
</section>
</div>
<div id="English" class="tab-pane fade" role="tabpanel" aria-labelledby="English-tab">
<p>Simulations were performed to outline the approximate analysis methodology. For your better understanding, I will briefly describe the simulation flow diagram. The simulation was conducted in 9 steps.</p>
<section id="settings-for-simulation" class="level4" data-number="3.0.1">
<h4 data-number="3.0.1" class="anchored" data-anchor-id="settings-for-simulation"><span class="header-section-number">3.0.1</span> Settings for Simulation</h4>
<p><strong>Data Set Size Setting</strong></p>
<p>Some of the setting values ​​required for simulation are set as global variables so that they can be freely called in the later scripts. (see <a href="#sec-global_variables" class="quarto-xref">Section&nbsp;2.2</a>)</p>
<p><strong>Categorial Data Setting</strong></p>
<p>I first set the dimensions of my high-dimensional data by setting the sample size and number of variables, then I created categorical data by choosing a well-known distribution, a distribution I determined, or a distribution that occurred randomly set. (see <a href="#sec-global_variables" class="quarto-xref">Section&nbsp;2.2</a>, <a href="#sec-functions" class="quarto-xref">Section&nbsp;2.3</a>, and <a href="#sec-simulation" class="quarto-xref">Section&nbsp;2.4</a>)</p>
<p><strong>Sex Variable Setting</strong></p>
<p>The data of the sex variable were generated through <span class="math inline">\(X \sim \text{Bernoulli}(0.5)\)</span>. (<a href="#sec-simulation" class="quarto-xref">Section&nbsp;2.4</a>)</p>
<p><strong>Genotype Variable Setting</strong></p>
<p>Personally, I have not yet figured out how to generate data for genotype (categorical) variables statistically, so further research is needed. However, since the genetic influence on the disease should be reflected, I tried to reflect the distribution of outcome variables and well-known distribution of the genotypes, APOE <a href="https://en.wikipedia.org/wiki/Apolipoprotein_E">(from Wiki)</a>. To make an association between the two variables, the proportions of each variable were set as marginal distirubtion and the joint distribution of the two variables was calculated to generate genotype data. (<a href="#sec-functions" class="quarto-xref">Section&nbsp;2.3</a> and <a href="#sec-simulation" class="quarto-xref">Section&nbsp;2.4</a>)</p>
<p><strong>Metabolite Data Setting</strong></p>
<p>As a setting for generating high-dimensional metabolomic data, a covariance matrix generated by random numbers is generated to create high-dimensional and mutually correlated metabolites within a group, which is used as input in the MVN (Multivariate Normal Distribution) function, and for each group, the metabolites’ effect (or weight) toward the outcome variable is also set to be randomly generated with a random number. At this time, the range of numbers randomly generated by random numbers was arbitrarily limited by myself. A seed number was fixed for reproducibility. (<a href="#sec-simulation" class="quarto-xref">Section&nbsp;2.4</a>)</p>
<p><strong>Outcome Variable Setting</strong></p>
<p>A score matrix was created through the matrix multiplication of the data created by MVN and a pre-made weight matrix with the probability values of samples that were created using the Logit Link. Based on the probability value of each sample, a binary outcome variable representing disease status information was created through <span class="math inline">\(X \sim \text{Bernoulli}(p)\)</span>, (where <span class="math inline">\(p\)</span> means the probability value of each sample). (<a href="#sec-simulation" class="quarto-xref">Section&nbsp;2.4</a>)</p>
<p><strong>Age Variable Setting</strong></p>
<p>Since the Age variable is a important factor related to dementia biologically and medically, the metabolite best explained as an Outcome variable was selected and converted into an age scale using min-max normalization by setting the youngest to 65 and the oldest to 105. (<a href="#sec-functions" class="quarto-xref">Section&nbsp;2.3</a> and <a href="#sec-simulation" class="quarto-xref">Section&nbsp;2.4</a>)</p>
<p><strong>Merge All Data</strong></p>
<p>Each variable created through simulation was merged into a data frame. (<a href="#sec-simulation" class="quarto-xref">Section&nbsp;2.4</a>)</p>
<p><strong>Analytics &amp; Conclusion</strong></p>
<p>The analysis part will not be discussed in detail in this data preparation section, but will be covered in detail in the EDA, Statistical Approaches and ML Approaches section. Briefly, I describe a method to identify the variables most associated with the outcome variable by selecting metabolites that have a statistically significant relationship with the outcome variable and comparing how similar the results are to those obtained through machine learning.</p>
</section>
<section id="simulation-1" class="level2" data-number="4">
<h2 data-number="4" class="anchored" data-anchor-id="simulation-1"><span class="header-section-number">4</span> Simulation</h2>
<section id="package-loading-and-option-settings-1" class="level3" data-number="4.1">
<h3 data-number="4.1" class="anchored" data-anchor-id="package-loading-and-option-settings-1"><span class="header-section-number">4.1</span> Package Loading and Option Settings</h3>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(janitor)) <span class="fu">install.packages</span>(<span class="st">"janitor"</span>) </span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(tidyverse)) <span class="fu">install.packages</span>(<span class="st">"tidyverse"</span>) </span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(tidymodels)) <span class="fu">install.packages</span>(<span class="st">"tidymodels"</span>) </span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(glmnet)) <span class="fu">install.packages</span>(<span class="st">"glmnet"</span>) </span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(MASS)) <span class="fu">install.packages</span>(<span class="st">"MASS"</span>) </span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(ggpubr)) <span class="fu">install.packages</span>(<span class="st">"ggpubr"</span>) </span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(car)) <span class="fu">install.packages</span>(<span class="st">"car"</span>) </span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(mixOmics)) <span class="fu">install.packages</span>(<span class="st">"mixOmics"</span>) </span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(janitor)</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidymodels)</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmnet)</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MASS)</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr) </span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(car) </span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mixOmics)</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">20230103</span>) </span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">message=</span><span class="cn">FALSE</span>,<span class="at">warning=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="global-variables" class="level3" data-number="4.2">
<h3 data-number="4.2" class="anchored" data-anchor-id="global-variables"><span class="header-section-number">4.2</span> Global Variables</h3>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># the number of samples</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>sample_size <span class="ot">&lt;-</span> <span class="dv">500</span> <span class="co">#1000</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co"># the number of predictors</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>predictor_size <span class="ot">&lt;-</span> <span class="dv">1000</span> <span class="co">#5000</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="co"># the number of groups</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>group_size <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="dv">6</span><span class="sc">:</span><span class="dv">10</span>,<span class="dv">1</span>) <span class="co"># at least more than 6, the number of the genotypes</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co"># the number of predictors truly associated with a response variable</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>significant_predictors <span class="ot">&lt;-</span> <span class="fu">floor</span>(predictor_size<span class="sc">*</span><span class="fu">sample</span>((<span class="dv">50</span><span class="sc">:</span><span class="dv">100</span>)<span class="sc">/</span><span class="dv">1000</span>,<span class="dv">1</span>)) </span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="do">## set the predictors associated with an outcome</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="do">### the number of predictors positively associated with an outcome</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="do">### the number of predictors negatively associated with an outcome</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>positively_associated_predictors<span class="ot">&lt;-</span><span class="fu">floor</span>(significant_predictors<span class="sc">*</span><span class="fl">0.4</span>) </span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>negatively_associated_predictors<span class="ot">&lt;-</span>significant_predictors<span class="sc">-</span>positively_associated_predictors </span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a><span class="do">## set the proportion of the groups in which the predictors are correlated with one another</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a><span class="do">### randomly sampling proportions to become their sum equal to 1</span></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>group_proportion_list<span class="ot">&lt;-</span><span class="fu">sample</span>(<span class="fu">seq</span>(<span class="dv">1</span>,<span class="dv">1</span><span class="sc">+</span><span class="dv">2</span><span class="sc">*</span>(<span class="dv">100</span><span class="sc">-</span>group_size)<span class="sc">/</span>group_size,</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>                            <span class="at">by=</span><span class="dv">2</span><span class="sc">*</span>(<span class="dv">100</span><span class="sc">-</span>group_size)<span class="sc">/</span>(group_size<span class="sc">*</span>(group_size<span class="dv">-1</span>)))<span class="sc">/</span><span class="dv">100</span>,</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>                        group_size,<span class="at">replace=</span><span class="cn">FALSE</span>)<span class="sc">%&gt;%</span><span class="fu">round</span>(<span class="dv">3</span>) </span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(group_proportion_list)<span class="ot">&lt;-</span><span class="fu">paste0</span>(<span class="st">"group"</span>,<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(group_proportion_list))</span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a><span class="do">### initialize a matrix with a size as sample_size by predictor_size</span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>predictor_matrix <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">0</span>, <span class="at">ncol =</span> predictor_size, <span class="at">nrow =</span> sample_size)</span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a><span class="do">### initialize a data frame and assign meta information used to generate simulated data</span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>group_meta_data<span class="ot">&lt;-</span></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(</span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">group_name=</span><span class="fu">c</span>(<span class="fu">names</span>(group_proportion_list)),</span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a>        <span class="at">proportion=</span>group_proportion_list)<span class="sc">%&gt;%</span></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mutate</span>(</span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a>            <span class="co"># the number of predictors within each group </span></span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>            <span class="at">group_n=</span>(predictor_size<span class="sc">*</span>group_proportion_list)<span class="sc">%&gt;%</span><span class="fu">round</span>(<span class="dv">0</span>),</span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a>            <span class="co"># the 1st index of predictors in each group </span></span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a>            <span class="at">first_index=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="fu">cumsum</span>(group_n[<span class="sc">-</span><span class="fu">length</span>(group_proportion_list)])<span class="sc">+</span><span class="dv">1</span>), </span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a>            <span class="co"># the last index of predictors in each group</span></span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a>            <span class="at">last_index=</span><span class="fu">cumsum</span>(group_n),</span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a>            <span class="co"># within-group correlations among the within-group predictors </span></span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a>            <span class="at">group_correlation=</span><span class="fu">sample</span>((<span class="dv">0</span><span class="sc">:</span><span class="dv">700</span>)<span class="sc">/</span><span class="dv">1000</span>,<span class="fu">length</span>(group_proportion_list),<span class="at">replace=</span><span class="cn">TRUE</span>), </span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a>            <span class="co"># effect of each group on an outcome variable</span></span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a>            <span class="at">group_effect=</span><span class="fu">sample</span>((<span class="sc">-</span><span class="dv">40</span><span class="sc">:</span><span class="dv">30</span>)<span class="sc">/</span><span class="dv">100</span>,<span class="fu">length</span>(group_proportion_list),<span class="at">replace=</span><span class="cn">TRUE</span>)) </span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a><span class="do">### set a group effect as 0.7 into a group with the smallest group number </span></span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a>group_meta_data[<span class="fu">which.min</span>(group_meta_data[,<span class="st">"group_n"</span>]),<span class="st">"group_effect"</span>]<span class="ot">&lt;-</span><span class="fl">0.7</span></span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a><span class="do">### set a group effect as -0.5 into a group with the second smallest group number </span></span>
<span id="cb7-44"><a href="#cb7-44" aria-hidden="true" tabindex="-1"></a>group_meta_data[group_meta_data[,<span class="st">"group_n"</span>]<span class="sc">==</span>(<span class="fu">sort</span>(group_meta_data[,<span class="st">"group_n"</span>])[<span class="dv">2</span>]),<span class="st">"group_effect"</span>]<span class="ot">&lt;-</span>(<span class="sc">-</span><span class="fl">0.5</span>)</span>
<span id="cb7-45"><a href="#cb7-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-46"><a href="#cb7-46" aria-hidden="true" tabindex="-1"></a><span class="co"># initialize a data matrix to assign simulated values</span></span>
<span id="cb7-47"><a href="#cb7-47" aria-hidden="true" tabindex="-1"></a><span class="do">## add some noise to data</span></span>
<span id="cb7-48"><a href="#cb7-48" aria-hidden="true" tabindex="-1"></a>data<span class="ot">&lt;-</span><span class="fu">matrix</span>(<span class="fu">rnorm</span>(sample_size<span class="sc">*</span>predictor_size,<span class="at">mean=</span><span class="dv">0</span>,<span class="at">sd=</span><span class="fl">0.05</span>), </span>
<span id="cb7-49"><a href="#cb7-49" aria-hidden="true" tabindex="-1"></a>             <span class="at">nrow =</span> sample_size, <span class="at">ncol =</span> predictor_size)</span>
<span id="cb7-50"><a href="#cb7-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-51"><a href="#cb7-51" aria-hidden="true" tabindex="-1"></a><span class="co"># initialize a covariance matrix to assign simulated values</span></span>
<span id="cb7-52"><a href="#cb7-52" aria-hidden="true" tabindex="-1"></a>covariance_matrix<span class="ot">&lt;-</span><span class="fu">matrix</span>(<span class="fu">rnorm</span>(predictor_size<span class="sc">*</span>predictor_size,<span class="at">mean=</span><span class="dv">0</span>,<span class="at">sd=</span><span class="fl">0.05</span>),</span>
<span id="cb7-53"><a href="#cb7-53" aria-hidden="true" tabindex="-1"></a>                          <span class="at">nrow=</span>predictor_size, <span class="at">ncol=</span>predictor_size)</span>
<span id="cb7-54"><a href="#cb7-54" aria-hidden="true" tabindex="-1"></a>beta_coefficients <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(predictor_size,<span class="dv">0</span>,<span class="fl">0.05</span>)</span>
<span id="cb7-55"><a href="#cb7-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-56"><a href="#cb7-56" aria-hidden="true" tabindex="-1"></a>answer_list<span class="ot">&lt;-</span><span class="fu">list</span>(</span>
<span id="cb7-57"><a href="#cb7-57" aria-hidden="true" tabindex="-1"></a>    <span class="st">'sample size'</span><span class="ot">=</span>sample_size,</span>
<span id="cb7-58"><a href="#cb7-58" aria-hidden="true" tabindex="-1"></a>    <span class="st">'predictor size'</span><span class="ot">=</span>predictor_size,</span>
<span id="cb7-59"><a href="#cb7-59" aria-hidden="true" tabindex="-1"></a>    <span class="st">'group size'</span><span class="ot">=</span>group_size,</span>
<span id="cb7-60"><a href="#cb7-60" aria-hidden="true" tabindex="-1"></a>    <span class="st">'significant predictors'</span><span class="ot">=</span>significant_predictors,</span>
<span id="cb7-61"><a href="#cb7-61" aria-hidden="true" tabindex="-1"></a>    <span class="st">'positively associated predictors'</span><span class="ot">=</span>positively_associated_predictors,</span>
<span id="cb7-62"><a href="#cb7-62" aria-hidden="true" tabindex="-1"></a>    <span class="st">'negatively associated predictors'</span><span class="ot">=</span>negatively_associated_predictors,</span>
<span id="cb7-63"><a href="#cb7-63" aria-hidden="true" tabindex="-1"></a>    <span class="st">'group proportion list'</span><span class="ot">=</span>group_proportion_list,</span>
<span id="cb7-64"><a href="#cb7-64" aria-hidden="true" tabindex="-1"></a>    <span class="st">'group meta data'</span><span class="ot">=</span>group_meta_data,</span>
<span id="cb7-65"><a href="#cb7-65" aria-hidden="true" tabindex="-1"></a>    <span class="st">'data noise intensity'</span><span class="ot">=</span><span class="fu">c</span>(<span class="st">'distribution'</span><span class="ot">=</span><span class="st">'rnorm'</span>,<span class="st">'mean'</span><span class="ot">=</span><span class="dv">0</span>,<span class="st">'sd'</span><span class="ot">=</span><span class="fl">0.05</span>),</span>
<span id="cb7-66"><a href="#cb7-66" aria-hidden="true" tabindex="-1"></a>    <span class="st">'covariance noise intensity'</span><span class="ot">=</span><span class="fu">c</span>(<span class="st">'distribution'</span><span class="ot">=</span><span class="st">'rnorm'</span>,<span class="st">'mean'</span><span class="ot">=</span><span class="dv">0</span>,<span class="st">'sd'</span><span class="ot">=</span><span class="fl">0.05</span>),</span>
<span id="cb7-67"><a href="#cb7-67" aria-hidden="true" tabindex="-1"></a>    <span class="st">'effect noise intensity on response'</span><span class="ot">=</span><span class="fu">c</span>(<span class="st">'distribution'</span><span class="ot">=</span><span class="st">'rnorm'</span>,<span class="st">'mean'</span><span class="ot">=</span><span class="dv">0</span>,<span class="st">'sd'</span><span class="ot">=</span><span class="fl">0.05</span>),</span>
<span id="cb7-68"><a href="#cb7-68" aria-hidden="true" tabindex="-1"></a>    <span class="st">'link function between the response and predictors'</span> <span class="ot">=</span> <span class="st">'canonical logit link function'</span>,</span>
<span id="cb7-69"><a href="#cb7-69" aria-hidden="true" tabindex="-1"></a>    <span class="st">'link function noise intensity'</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'distribution'</span><span class="ot">=</span><span class="st">'rnorm'</span>,<span class="st">'mean'</span><span class="ot">=</span><span class="dv">0</span>,<span class="st">'sd'</span><span class="ot">=</span><span class="fl">0.05</span>),</span>
<span id="cb7-70"><a href="#cb7-70" aria-hidden="true" tabindex="-1"></a>    <span class="st">'age_distirbution'</span><span class="ot">=</span><span class="st">'used data of a variable with the highest effect on outcome'</span>,</span>
<span id="cb7-71"><a href="#cb7-71" aria-hidden="true" tabindex="-1"></a>    <span class="st">'sex_distribution'</span><span class="ot">=</span><span class="st">'rbinom(n=sample size,p=0.5)'</span>,</span>
<span id="cb7-72"><a href="#cb7-72" aria-hidden="true" tabindex="-1"></a>    <span class="st">'genotype_distirbution'</span><span class="ot">=</span><span class="fu">c</span>(<span class="st">'e2'</span> <span class="ot">=</span> <span class="st">'8.4%'</span>,<span class="st">'e3'</span> <span class="ot">=</span> <span class="st">'77.9%'</span>,<span class="st">'e4'</span> <span class="ot">=</span> <span class="st">'13.7%'</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="functions" class="level3" data-number="4.3">
<h3 data-number="4.3" class="anchored" data-anchor-id="functions"><span class="header-section-number">4.3</span> Functions</h3>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Function List</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>scale_function<span class="ot">=</span><span class="cf">function</span>(<span class="at">vector=</span>x,<span class="at">min=</span>a,<span class="at">max=</span>b,<span class="at">method=</span><span class="st">"customized"</span>){</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(method<span class="sc">==</span><span class="st">"min-max"</span>){</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>        result<span class="ot">=</span>(vector<span class="sc">-</span><span class="fu">min</span>(vector))<span class="sc">/</span>(<span class="fu">max</span>(vector)<span class="sc">-</span><span class="fu">min</span>(vector))</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>    }<span class="cf">else</span> <span class="cf">if</span>(method<span class="sc">==</span><span class="st">"customized"</span>){</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>        result<span class="ot">=</span>(max<span class="sc">-</span>min)<span class="sc">*</span>(vector<span class="sc">-</span><span class="fu">min</span>(vector))<span class="sc">/</span>(<span class="fu">max</span>(vector)<span class="sc">-</span><span class="fu">min</span>(vector))<span class="sc">+</span>min</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>    }<span class="cf">else</span>{</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>        result<span class="ot">=</span>(vector<span class="sc">-</span><span class="fu">mean</span>(vector))<span class="sc">/</span><span class="fu">sd</span>(vector)</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(result)</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>age_data_generator<span class="ot">=</span><span class="cf">function</span>(in_data,in_response,<span class="at">fun=</span>scale_function){</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>    <span class="co"># this function generates a age (continuous) data that are statistically associated </span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>    <span class="co"># with a simulated variable as designed above.</span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>    <span class="do">## conduct t test with the response and each variable generated by multivariate normal distributions.</span></span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>    <span class="do">## search a variable with the largest difference in mean between the two groups or with the lowest p value</span></span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>    <span class="do">## In this case, I will pick the former one. </span></span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>    <span class="do">## (I don't care about the multiple testing problems for now)</span></span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a>    temp_df<span class="ot">=</span><span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="at">ncol=</span><span class="dv">5</span>)) <span class="co"># initialize an empty data frame</span></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(in_data)){</span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a>        temp_df[i,]<span class="ot">=</span><span class="fu">c</span>(</span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a>            <span class="fu">names</span>(in_data)[i],</span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a>            <span class="fu">t.test</span>(in_data[,i]<span class="sc">~</span>in_response)<span class="sc">$</span>estimate[<span class="dv">1</span>],</span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a>            <span class="fu">t.test</span>(in_data[,i]<span class="sc">~</span>in_response)<span class="sc">$</span>estimate[<span class="dv">2</span>],</span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a>            <span class="fu">t.test</span>(in_data[,i]<span class="sc">~</span>in_response)<span class="sc">$</span>estimate[<span class="dv">2</span>]<span class="sc">-</span><span class="fu">t.test</span>(data[,i]<span class="sc">~</span>response)<span class="sc">$</span>estimate[<span class="dv">1</span>],</span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a>            <span class="fu">t.test</span>(in_data[,i]<span class="sc">~</span>in_response)<span class="sc">$</span>p.value)</span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb8-31"><a href="#cb8-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(temp_df)<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">'metabolite'</span>,<span class="st">'mean_neg'</span>,<span class="st">'mean_pos'</span>,<span class="st">'mean_diff'</span>,<span class="st">'p.value'</span>)</span>
<span id="cb8-32"><a href="#cb8-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-33"><a href="#cb8-33" aria-hidden="true" tabindex="-1"></a>    <span class="do">## search a variable with the largest difference in mean</span></span>
<span id="cb8-34"><a href="#cb8-34" aria-hidden="true" tabindex="-1"></a>    strong_metabolite<span class="ot">&lt;-</span></span>
<span id="cb8-35"><a href="#cb8-35" aria-hidden="true" tabindex="-1"></a>        temp_df<span class="sc">%&gt;%</span></span>
<span id="cb8-36"><a href="#cb8-36" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mutate</span>(</span>
<span id="cb8-37"><a href="#cb8-37" aria-hidden="true" tabindex="-1"></a>            <span class="at">mean_neg=</span><span class="fu">as.numeric</span>(mean_neg),</span>
<span id="cb8-38"><a href="#cb8-38" aria-hidden="true" tabindex="-1"></a>            <span class="at">mean_pos=</span><span class="fu">as.numeric</span>(mean_pos),</span>
<span id="cb8-39"><a href="#cb8-39" aria-hidden="true" tabindex="-1"></a>            <span class="at">mean_diff=</span><span class="fu">as.numeric</span>(mean_diff),</span>
<span id="cb8-40"><a href="#cb8-40" aria-hidden="true" tabindex="-1"></a>            <span class="at">p.value=</span><span class="fu">as.numeric</span>(p.value),</span>
<span id="cb8-41"><a href="#cb8-41" aria-hidden="true" tabindex="-1"></a>            <span class="at">abs_mean_diff=</span><span class="fu">abs</span>(mean_diff))<span class="sc">%&gt;%</span></span>
<span id="cb8-42"><a href="#cb8-42" aria-hidden="true" tabindex="-1"></a>        <span class="fu">filter</span>(abs_mean_diff<span class="sc">==</span><span class="fu">max</span>(abs_mean_diff))<span class="sc">%&gt;%</span></span>
<span id="cb8-43"><a href="#cb8-43" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(metabolite)<span class="sc">%&gt;%</span>pull</span>
<span id="cb8-44"><a href="#cb8-44" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb8-45"><a href="#cb8-45" aria-hidden="true" tabindex="-1"></a>    <span class="do">## generate age data with min max normalization</span></span>
<span id="cb8-46"><a href="#cb8-46" aria-hidden="true" tabindex="-1"></a>    age_data<span class="ot">&lt;-</span></span>
<span id="cb8-47"><a href="#cb8-47" aria-hidden="true" tabindex="-1"></a>        data<span class="sc">%&gt;%</span></span>
<span id="cb8-48"><a href="#cb8-48" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(strong_metabolite)<span class="sc">%&gt;%</span></span>
<span id="cb8-49"><a href="#cb8-49" aria-hidden="true" tabindex="-1"></a>        <span class="fu">scale_function</span>(<span class="at">vector=</span>.,<span class="at">min=</span><span class="dv">65</span>,<span class="at">max=</span><span class="dv">105</span>,<span class="at">method=</span><span class="st">"customized"</span>)<span class="sc">%&gt;%</span></span>
<span id="cb8-50"><a href="#cb8-50" aria-hidden="true" tabindex="-1"></a>        <span class="fu">rename</span>(<span class="at">age=</span><span class="dv">1</span>)<span class="sc">%&gt;%</span><span class="fu">round</span>(<span class="dv">0</span>)</span>
<span id="cb8-51"><a href="#cb8-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(age_data)</span>
<span id="cb8-52"><a href="#cb8-52" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-53"><a href="#cb8-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-54"><a href="#cb8-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-55"><a href="#cb8-55" aria-hidden="true" tabindex="-1"></a>genotype_data_generator<span class="ot">=</span><span class="cf">function</span>(<span class="at">in_response=</span>response,<span class="at">fun=</span>scale_function){</span>
<span id="cb8-56"><a href="#cb8-56" aria-hidden="true" tabindex="-1"></a>    <span class="co"># this function generates a genotype (categorical) data </span></span>
<span id="cb8-57"><a href="#cb8-57" aria-hidden="true" tabindex="-1"></a>    <span class="co"># that are jointly and statistically associated with a continuous data and a binary data </span></span>
<span id="cb8-58"><a href="#cb8-58" aria-hidden="true" tabindex="-1"></a>    <span class="co"># (I am not so sure if I can generate data that are statistically associated </span></span>
<span id="cb8-59"><a href="#cb8-59" aria-hidden="true" tabindex="-1"></a>    <span class="co"># with some fake metabolite data. But, I will give it a try).</span></span>
<span id="cb8-60"><a href="#cb8-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-61"><a href="#cb8-61" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Declare the marginal proportions </span></span>
<span id="cb8-62"><a href="#cb8-62" aria-hidden="true" tabindex="-1"></a>    <span class="do">## for binary (affected vs unaffected) and a genotype (categorical) data, respectively</span></span>
<span id="cb8-63"><a href="#cb8-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-64"><a href="#cb8-64" aria-hidden="true" tabindex="-1"></a>    <span class="do">##the simulated proportion for the disease vs non-disease cases</span></span>
<span id="cb8-65"><a href="#cb8-65" aria-hidden="true" tabindex="-1"></a>    binary_proportion<span class="ot">&lt;-</span><span class="fu">as.numeric</span>(<span class="fu">table</span>(in_response)<span class="sc">/</span>sample_size) </span>
<span id="cb8-66"><a href="#cb8-66" aria-hidden="true" tabindex="-1"></a>    genotype_proportion<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="fl">0.084</span>,<span class="fl">0.779</span>,<span class="fl">0.137</span>) <span class="co"># the known proportion of APOE genotypes from Wiki</span></span>
<span id="cb8-67"><a href="#cb8-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-68"><a href="#cb8-68" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Declare the joint proportion predictor matrix</span></span>
<span id="cb8-69"><a href="#cb8-69" aria-hidden="true" tabindex="-1"></a>    joint_proportion<span class="ot">&lt;-</span><span class="fu">matrix</span>(</span>
<span id="cb8-70"><a href="#cb8-70" aria-hidden="true" tabindex="-1"></a>        <span class="fu">c</span>(binary_proportion[<span class="dv">1</span>]<span class="sc">*</span>genotype_proportion, <span class="co"># for the unaffected cases</span></span>
<span id="cb8-71"><a href="#cb8-71" aria-hidden="true" tabindex="-1"></a>        binary_proportion[<span class="dv">2</span>]<span class="sc">*</span>genotype_proportion),  <span class="co"># for the affected cases</span></span>
<span id="cb8-72"><a href="#cb8-72" aria-hidden="true" tabindex="-1"></a>        <span class="at">ncol=</span><span class="dv">2</span>, <span class="at">byrow=</span><span class="cn">FALSE</span>)</span>
<span id="cb8-73"><a href="#cb8-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-74"><a href="#cb8-74" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Generate the genotype (catogrical) data</span></span>
<span id="cb8-75"><a href="#cb8-75" aria-hidden="true" tabindex="-1"></a>    genotype_data <span class="ot">=</span> <span class="fu">numeric</span>(sample_size) <span class="co"># initialize a vector </span></span>
<span id="cb8-76"><a href="#cb8-76" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>sample_size) {</span>
<span id="cb8-77"><a href="#cb8-77" aria-hidden="true" tabindex="-1"></a>        genotype_data[i] <span class="ot">=</span> <span class="fu">sample</span>(</span>
<span id="cb8-78"><a href="#cb8-78" aria-hidden="true" tabindex="-1"></a>            <span class="fu">c</span>(<span class="st">'e2'</span>,<span class="st">'e3'</span>,<span class="st">'e4'</span>),</span>
<span id="cb8-79"><a href="#cb8-79" aria-hidden="true" tabindex="-1"></a>             <span class="dv">1</span>, </span>
<span id="cb8-80"><a href="#cb8-80" aria-hidden="true" tabindex="-1"></a>             <span class="at">prob=</span>joint_proportion[,<span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">'neg'</span>,in_response[i]),<span class="dv">1</span>,<span class="dv">2</span>)])</span>
<span id="cb8-81"><a href="#cb8-81" aria-hidden="true" tabindex="-1"></a>    }    </span>
<span id="cb8-82"><a href="#cb8-82" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(genotype_data)</span>
<span id="cb8-83"><a href="#cb8-83" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="simulation-2" class="level3" data-number="4.4">
<h3 data-number="4.4" class="anchored" data-anchor-id="simulation-2"><span class="header-section-number">4.4</span> Simulation</h3>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># generate simulation data using multivariate normal distribution</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(group_meta_data)) {</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>    group_range <span class="ot">&lt;-</span> group_meta_data[i, <span class="st">"first_index"</span>]<span class="sc">:</span>group_meta_data[i, <span class="st">"last_index"</span>]</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (j <span class="cf">in</span> group_range){</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span>(k <span class="cf">in</span> group_range){</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>        covariance_matrix[j, k] <span class="ot">&lt;-</span> group_meta_data[i, <span class="st">"group_correlation"</span>]</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>    <span class="co">#covariance_matrix[group_range, group_range]+group_meta_data[i, "group_correlation"]    </span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">diag</span>(covariance_matrix) <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>    data[, group_range] <span class="ot">&lt;-</span> </span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mvrnorm</span>(<span class="at">n =</span> sample_size, </span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>                <span class="at">mu =</span> <span class="fu">rep</span>(<span class="dv">0</span>,group_meta_data[i,<span class="st">"group_n"</span>]),</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>                <span class="at">Sigma =</span> covariance_matrix[group_range, group_range])</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>    data<span class="ot">=</span><span class="fu">as.data.frame</span>(data)</span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>    beta_coefficients[group_meta_data[i, <span class="st">"first_index"</span>]<span class="sc">:</span>group_meta_data[i, <span class="st">"last_index"</span>]] <span class="ot">&lt;-</span></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>        beta_coefficients[group_meta_data[i, <span class="st">"first_index"</span>]<span class="sc">:</span>group_meta_data[i, <span class="st">"last_index"</span>]]<span class="sc">+</span></span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>        group_meta_data[i,<span class="st">"group_effect"</span>]</span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>    predictor_names<span class="ot">&lt;-</span><span class="fu">paste0</span>(group_meta_data[i,<span class="st">"group_name"</span>],<span class="st">"_"</span>,<span class="dv">1</span><span class="sc">:</span>group_meta_data[i,<span class="st">"group_n"</span>])</span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(beta_coefficients)[group_meta_data[i, <span class="st">"first_index"</span>]<span class="sc">:</span>group_meta_data[i, <span class="st">"last_index"</span>]] <span class="ot">&lt;-</span> predictor_names</span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(data)[group_meta_data[i, <span class="st">"first_index"</span>]<span class="sc">:</span>group_meta_data[i, <span class="st">"last_index"</span>]]<span class="ot">&lt;-</span>predictor_names</span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>score<span class="ot">=</span><span class="fu">as.matrix</span>(data)<span class="sc">%*%</span>beta_coefficients <span class="co"># score of each sample</span></span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a><span class="co"># logistic function to get a probability, intercept = 0, </span></span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a><span class="do">## set probabilities-0.2 to apply noise and negative probabilities into 0</span></span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a>probabilities <span class="ot">&lt;-</span> </span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a>    ((<span class="dv">1</span><span class="sc">/</span>(<span class="dv">1</span><span class="sc">+</span><span class="fu">exp</span>(<span class="sc">-</span>(<span class="dv">0</span><span class="sc">+</span>score))))<span class="sc">-</span><span class="fu">rnorm</span>(sample_size,<span class="at">m=</span><span class="fl">0.2</span>,<span class="at">sd=</span><span class="fl">0.05</span>))<span class="sc">%&gt;%</span></span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ifelse</span>(.<span class="sc">&gt;</span><span class="dv">1</span>,<span class="dv">1</span>,.)<span class="sc">%&gt;%</span></span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ifelse</span>(.<span class="sc">&lt;</span><span class="dv">0</span>,<span class="dv">0</span>,.)</span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a>response <span class="ot">&lt;-</span></span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ifelse</span>(<span class="fu">rbinom</span>(sample_size, <span class="dv">1</span>, probabilities)<span class="sc">==</span><span class="dv">0</span>,<span class="st">'negative'</span>,<span class="st">'positive'</span>)<span class="sc">%&gt;%</span></span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">factor</span>(.,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">'negative'</span>,<span class="st">'positive'</span>))</span>
<span id="cb9-37"><a href="#cb9-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-38"><a href="#cb9-38" aria-hidden="true" tabindex="-1"></a><span class="co"># simulate sex data</span></span>
<span id="cb9-39"><a href="#cb9-39" aria-hidden="true" tabindex="-1"></a>sex_data <span class="ot">&lt;-</span> <span class="fu">rbinom</span>(sample_size,<span class="dv">1</span>,<span class="fl">0.5</span>)</span>
<span id="cb9-40"><a href="#cb9-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-41"><a href="#cb9-41" aria-hidden="true" tabindex="-1"></a><span class="co"># simulate age data</span></span>
<span id="cb9-42"><a href="#cb9-42" aria-hidden="true" tabindex="-1"></a>age_data <span class="ot">&lt;-</span> <span class="fu">age_data_generator</span>(<span class="at">in_data=</span>data,<span class="at">in_response=</span>response)</span>
<span id="cb9-43"><a href="#cb9-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-44"><a href="#cb9-44" aria-hidden="true" tabindex="-1"></a><span class="co"># simulate genotype data</span></span>
<span id="cb9-45"><a href="#cb9-45" aria-hidden="true" tabindex="-1"></a>genotype_data <span class="ot">&lt;-</span> <span class="fu">genotype_data_generator</span>(<span class="at">in_response=</span>response)</span>
<span id="cb9-46"><a href="#cb9-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-47"><a href="#cb9-47" aria-hidden="true" tabindex="-1"></a><span class="co"># merge the univariables</span></span>
<span id="cb9-48"><a href="#cb9-48" aria-hidden="true" tabindex="-1"></a>phenotype_data<span class="ot">&lt;-</span></span>
<span id="cb9-49"><a href="#cb9-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(</span>
<span id="cb9-50"><a href="#cb9-50" aria-hidden="true" tabindex="-1"></a>        <span class="at">id=</span><span class="dv">1</span><span class="sc">:</span>sample_size,</span>
<span id="cb9-51"><a href="#cb9-51" aria-hidden="true" tabindex="-1"></a>        <span class="at">outcome=</span>response,</span>
<span id="cb9-52"><a href="#cb9-52" aria-hidden="true" tabindex="-1"></a>        <span class="at">age=</span>age_data,</span>
<span id="cb9-53"><a href="#cb9-53" aria-hidden="true" tabindex="-1"></a>        <span class="at">sex=</span>sex_data,</span>
<span id="cb9-54"><a href="#cb9-54" aria-hidden="true" tabindex="-1"></a>        <span class="at">genotype=</span>genotype_data)</span>
<span id="cb9-55"><a href="#cb9-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-56"><a href="#cb9-56" aria-hidden="true" tabindex="-1"></a>all_data<span class="ot">=</span><span class="fu">inner_join</span>(</span>
<span id="cb9-57"><a href="#cb9-57" aria-hidden="true" tabindex="-1"></a>    phenotype_data,</span>
<span id="cb9-58"><a href="#cb9-58" aria-hidden="true" tabindex="-1"></a>    data<span class="sc">%&gt;%</span><span class="fu">mutate</span>(<span class="at">id=</span><span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>()),</span>
<span id="cb9-59"><a href="#cb9-59" aria-hidden="true" tabindex="-1"></a>    <span class="at">by=</span><span class="st">"id"</span>)</span>
<span id="cb9-60"><a href="#cb9-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-61"><a href="#cb9-61" aria-hidden="true" tabindex="-1"></a><span class="co">#write_rds(all_data,"./docs/data/llfs_simulated_data.rds")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="load-data-1" class="level3" data-number="4.5">
<h3 data-number="4.5" class="anchored" data-anchor-id="load-data-1"><span class="header-section-number">4.5</span> Load Data</h3>
<div class="cell">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load simulation data</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>simulated_data<span class="ot">&lt;-</span><span class="fu">read_rds</span>(datapath)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># simple data pre-processing</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>all_data<span class="ot">&lt;-</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>    simulated_data<span class="sc">%&gt;%</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">outcome=</span><span class="fu">factor</span>(outcome,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">"negative"</span>,<span class="st">"positive"</span>)),</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">sex=</span><span class="fu">ifelse</span>(sex<span class="sc">==</span><span class="dv">0</span>,<span class="st">"man"</span>,<span class="st">"woman"</span>),</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>      <span class="at">sex=</span><span class="fu">factor</span>(sex,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">"man"</span>,<span class="st">"woman"</span>)),</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>      <span class="at">genotype=</span><span class="fu">factor</span>(genotype,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">"e3"</span>,<span class="st">"e2"</span>,<span class="st">"e4"</span>))</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a><span class="co"># rename metabolite variables</span></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(all_data)[<span class="dv">6</span><span class="sc">:</span><span class="fu">ncol</span>(all_data)]<span class="ot">&lt;-</span><span class="fu">paste0</span>(<span class="st">"meta"</span>,<span class="dv">1</span><span class="sc">:</span>predictor_size)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="data-description-1" class="level2" data-number="5">
<h2 data-number="5" class="anchored" data-anchor-id="data-description-1"><span class="header-section-number">5</span> Data Description</h2>
<p>This data include 500 samples and 1005 variables:</p>
<ul>
<li>id: sample ID.</li>
<li>outcome: a disease status (positive, negative), positive is an affected status, negative is an unaffected status, and the reference group is positive.</li>
<li>age: age</li>
<li>sex: sex (man, woman) and the reference group is man.</li>
<li>genotype: APOE genotypes
<ul>
<li>the apolipoprotein <span class="math inline">\(\epsilon\)</span> (APOE) is a protein produced in the metabolic pathway of fats in mammals, a genotype of which seems to be related to Alzheimer’s disease (AD). APOE is polymorphic and has three major alleles, <span class="math inline">\(\epsilon 2\)</span> (e2), <span class="math inline">\(\epsilon 3\)</span>(e3), and <span class="math inline">\(\epsilon 4\)</span> (e4). The statistics of the polymorphism are 8.4% for e2, 77.9% for e3, and 13.7% for e4 in worldwide allel frequency, respectively. It is known that the e2, e3, and e4 allels are associated with the protective factor, the neutral one, and the risk one with regard to AD. However, this finding has not been replicated in a large population. Therefore, it is known that we do not know their true associations with AD in the true population. <a href="https://en.wikipedia.org/wiki/Apolipoprotein_E">(from Wiki)</a>
<ul>
<li>There are 6 combinations of the genotypes:
<ul>
<li>e2/e2</li>
<li>e2/e3</li>
<li>e2/e4</li>
<li>e3/e3</li>
<li>e3/e4</li>
<li>e4/e4</li>
</ul></li>
<li>In this study, I generated the 3 major alleles, e3, e4, e2</li>
</ul></li>
</ul></li>
<li>meta1 ~ meta1000: a list of metabolites that were blood-sampled from the APOE carriers.</li>
</ul>
</section>
</div>


</div>
</ul></section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("kmink3225\.netlify\.app");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>