<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.313">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Kwangmin Kim">
<meta name="dcterms.date" content="2022-12-20">

<title>Kwangmin Kim - Data Preparation</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>
<script src="../../../site_libs/quarto-diagram/mermaid.min.js"></script>
<script src="../../../site_libs/quarto-diagram/mermaid-init.js"></script>
<link href="../../../site_libs/quarto-diagram/mermaid.css" rel="stylesheet">

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a href="../../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../.././images/logo.png" alt="" class="navbar-logo">
    </a>
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">Kwangmin Kim</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../index.html">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../docs/CV/index.html">
 <span class="menu-text">CV</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="../../../docs/projects/index.html" aria-current="page">
 <span class="menu-text">Projects</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../docs/blog/index.html">
 <span class="menu-text">Blog</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../about.html">
 <span class="menu-text">Me</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/kmink3225"><i class="bi bi-github" role="img" aria-label="Github">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://www.linkedin.com/in/kwangmin-kim-a5241b200/"><i class="bi bi-linkedin" role="img" aria-label="Linkedin">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../docs/blog/index.html"><i class="bi bi-rss" role="img">
</i> 
 <span class="menu-text">Blog</span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">Data Preparation</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../docs/projects/index.html" class="sidebar-item-text sidebar-link">Projects</a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">LLFS</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../docs/projects/LLFS/project_description.html" class="sidebar-item-text sidebar-link">Description</a>
  </div>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">Self Project</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth2 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../docs/projects/LLFS/self_description.html" class="sidebar-item-text sidebar-link">Description</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../docs/projects/LLFS/data_preparation.html" class="sidebar-item-text sidebar-link active">Data Preparation</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../docs/projects/LLFS/eda.html" class="sidebar-item-text sidebar-link">EDA</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../docs/projects/LLFS/statistical_approach.html" class="sidebar-item-text sidebar-link">Statistical Approach</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../docs/projects/LLFS/ml_approach.html" class="sidebar-item-text sidebar-link">ML Approach</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../docs/projects/LLFS/report.html" class="sidebar-item-text sidebar-link">Conclusion</a>
  </div>
</li>
      </ul>
  </li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#to-be" id="toc-to-be" class="nav-link active" data-scroll-target="#to-be"><span class="toc-section-number">0.1</span>  To-Be</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content column-body" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block">Data Preparation</h1>
<p class="subtitle lead">Simulation</p>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Kwangmin Kim </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">December 20, 2022</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="to-be" class="level3" data-number="0.1">
<h3 data-number="0.1" class="anchored" data-anchor-id="to-be"><span class="header-section-number">0.1</span> To-Be</h3>
<div class="cell" data-fig-width="7.5">
<div class="cell-output-display">
<div>
<p>
</p><pre class="mermaid mermaid-js" data-tooltip-selector="#mermaid-tooltip-1">flowchart TB
    subgraph Simulation
        direction TB
        subgraph assign_frequency
        direction LR
        Assign_probabilities     
        end
        subgraph generate_multivariate_normal_distribution
            direction TB
            determine_sample_size ---
            determine_number_of_columns---
            create_covariate_correlation_matrix---
            apply_noise_to_the_covariance_matrix
        end
        subgraph create_weight_matrix
            direction LR
            apply_some_noise_to_the_covariance_matrix
        end
        subgraph calculate_score_matrix
            direction LR
            matrix_multiplication_covariance_matrix_weight_matrix---
            used_logistic_function
        end
        subgraph create_response_variable
            direction LR
            result_of_logistic_function
        end
        subgraph creation_some_discrete_variables
            direction LR       
        end
        subgraph join
            direction LR
            all_the_matrix_and_vector_created       
        end
        assign_frequency--&gt;generate_multivariate_normal_distribution--&gt;create_weight_matrix--&gt;
        calculate_score_matrix--&gt;create_response_variable--&gt;creation_some_discrete_variables--&gt;
        join
    end
    subgraph data_analytics
        direction LR
        exploratory_data_analysis---
        statistical_analysis---
        machine_learning
    end
    subgraph conclusion
        direction LR
    end
    Simulation--&gt;data_analytics--&gt;conclusion
</pre>
<div id="mermaid-tooltip-1" class="mermaidTooltip">

</div>
<p></p>
</div>
</div>
</div>
<ul class="nav nav-pills" id="language-tab" role="tablist">
<li class="nav-item" role="presentation">
<button class="nav-link active" id="Korean-tab" data-bs-toggle="tab" data-bs-target="#Korean" type="button" role="tab" aria-controls="Korean" aria-selected="true">
<p>Korean</p>
</button>
</li>
<li class="nav-item" role="presentation">
<button class="nav-link" id="English-tab" data-bs-toggle="tab" data-bs-target="#English" type="button" role="tab" aria-controls="knitr" aria-selected="false">
<p>English</p>
</button>
</li>
<div class="tab-content" id="language-tabcontent">

</div></ul></section>
<div id="Korean" class="tab-pane fade show active" role="tabpanel" aria-labelledby="Korean-tab">
<section id="준비중" class="level2" data-number="1">
<h2 data-number="1" class="anchored" data-anchor-id="준비중"><span class="header-section-number">1</span> 준비중</h2>
<p>please, read the English section first.</p>
</section>
</div>
<div id="English" class="tab-pane fade" role="tabpanel" aria-labelledby="English-tab">
<section id="data-preparation" class="level2" data-number="2">
<h2 data-number="2" class="anchored" data-anchor-id="data-preparation"><span class="header-section-number">2</span> Data Preparation</h2>
<section id="package-loading-and-option-settings" class="level3" data-number="2.1">
<h3 data-number="2.1" class="anchored" data-anchor-id="package-loading-and-option-settings"><span class="header-section-number">2.1</span> Package Loading and Option Settings</h3>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidymodels)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmnet)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MASS)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mixOmics)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">20221213</span>) </span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">message=</span><span class="cn">FALSE</span>,<span class="at">warning=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="data-simulation" class="level3" data-number="2.2">
<h3 data-number="2.2" class="anchored" data-anchor-id="data-simulation"><span class="header-section-number">2.2</span> Data Simulation</h3>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># the number of samples</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>sample_size <span class="ot">&lt;-</span> <span class="dv">1000</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co"># the number of predictors</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>predictor_size <span class="ot">&lt;-</span> <span class="dv">10000</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>group_size <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co"># the number of predictors truly associated with a response variable</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>significant_predictors <span class="ot">&lt;-</span> <span class="fu">floor</span>(predictor_size<span class="sc">*</span><span class="fu">sample</span>((<span class="dv">50</span><span class="sc">:</span><span class="dv">200</span>)<span class="sc">/</span><span class="dv">1000</span>,<span class="dv">1</span>)) </span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="do">## set the predictors associated with an outcome</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="do">### the number of predictors positively associated with an outcome</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="do">### the number of predictors negatively associated with an outcome</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>positively_associated_predictors<span class="ot">&lt;-</span><span class="fu">floor</span>(significant_predictors<span class="sc">*</span><span class="fl">0.4</span>) </span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>negatively_associated_predictors<span class="ot">&lt;-</span>significant_predictors<span class="sc">-</span>positively_associated_predictors </span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a><span class="do">## set correlated predictors within each group</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a><span class="do">### randomly sampling proportions of 10 correlated predictor groups </span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a><span class="do">### to become their sum equal to 1</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>proportion_list<span class="ot">&lt;-</span><span class="fu">sample</span>(<span class="fu">seq</span>(<span class="dv">1</span>,<span class="dv">1</span><span class="sc">+</span><span class="dv">2</span><span class="sc">*</span>(<span class="dv">100</span><span class="sc">-</span>group_size)<span class="sc">/</span>group_size,</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>                            <span class="at">by=</span><span class="dv">2</span><span class="sc">*</span>(<span class="dv">100</span><span class="sc">-</span>group_size)<span class="sc">/</span>(group_size<span class="sc">*</span>(group_size<span class="dv">-1</span>)))<span class="sc">/</span><span class="dv">100</span>,</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>                        group_size,<span class="at">replace=</span><span class="cn">FALSE</span>)<span class="sc">%&gt;%</span><span class="fu">round</span>(<span class="dv">3</span>) </span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(proportion_list)<span class="ot">&lt;-</span><span class="fu">paste0</span>(<span class="st">"group"</span>,<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(proportion_list))</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a><span class="do">### initialize a matrix with a size as sample_size by predictor_size</span></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>predictor_matrix <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">0</span>, <span class="at">ncol =</span> predictor_size, <span class="at">nrow =</span> sample_size)</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a><span class="do">### initialize a data frame and assign meta information used to generate simulated data</span></span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>meta_data<span class="ot">&lt;-</span></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">group_name=</span><span class="fu">c</span>(<span class="fu">names</span>(proportion_list)),</span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>               <span class="at">proportion=</span>proportion_list)<span class="sc">%&gt;%</span></span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">group_n=</span>(predictor_size<span class="sc">*</span>proportion_list)<span class="sc">%&gt;%</span><span class="fu">round</span>(<span class="dv">0</span>), <span class="co"># the within-group number of predictors</span></span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>           <span class="at">first_index=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="fu">cumsum</span>(group_n[<span class="sc">-</span><span class="fu">length</span>(proportion_list)])<span class="sc">+</span><span class="dv">1</span>), <span class="co"># the 1st index of predictors in each group</span></span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>           <span class="at">last_index=</span><span class="fu">cumsum</span>(group_n), <span class="co"># the last index of predictors in each group</span></span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>           <span class="at">group_correlation=</span><span class="fu">sample</span>((<span class="dv">0</span><span class="sc">:</span><span class="dv">700</span>)<span class="sc">/</span><span class="dv">1000</span>,<span class="fu">length</span>(proportion_list),<span class="at">replace=</span><span class="cn">TRUE</span>), <span class="co"># correlation among the within-group predictors</span></span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>           <span class="at">group_effect=</span><span class="fu">sample</span>((<span class="sc">-</span><span class="dv">2</span><span class="sc">:</span><span class="dv">2</span>)<span class="sc">/</span><span class="dv">10</span>,<span class="fu">length</span>(proportion_list),<span class="at">replace=</span><span class="cn">TRUE</span>)); <span class="co"># effect of each group on an outcome variable</span></span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>data<span class="ot">&lt;-</span><span class="fu">matrix</span>(<span class="fu">rnorm</span>(sample_size<span class="sc">*</span>predictor_size,<span class="at">mean=</span><span class="dv">0</span>,<span class="at">sd=</span><span class="fl">0.01</span>), </span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>             <span class="at">nrow =</span> sample_size, <span class="at">ncol =</span> predictor_size)</span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a>covariance_matrix<span class="ot">&lt;-</span><span class="fu">matrix</span>(<span class="fu">rnorm</span>(predictor_size<span class="sc">*</span>predictor_size,<span class="fl">0.15</span>,<span class="fl">0.05</span>),</span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>                          <span class="at">nrow=</span>predictor_size, <span class="at">ncol=</span>predictor_size)</span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a>beta_coefficients <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(predictor_size,<span class="dv">0</span>,<span class="fl">0.05</span>)</span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(meta_data)) {</span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a>    group_range <span class="ot">&lt;-</span> meta_data[i, <span class="st">"first_index"</span>]<span class="sc">:</span>meta_data[i, <span class="st">"last_index"</span>]</span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (j <span class="cf">in</span> group_range){</span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span>(k <span class="cf">in</span> group_range){</span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a>        covariance_matrix[j, k] <span class="ot">&lt;-</span> meta_data[i, <span class="st">"group_correlation"</span>]</span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb2-49"><a href="#cb2-49" aria-hidden="true" tabindex="-1"></a>    <span class="co">#covariance_matrix[group_range, group_range]+meta_data[i, "group_correlation"]    </span></span>
<span id="cb2-50"><a href="#cb2-50" aria-hidden="true" tabindex="-1"></a>    <span class="fu">diag</span>(covariance_matrix) <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb2-51"><a href="#cb2-51" aria-hidden="true" tabindex="-1"></a>    data[, group_range] <span class="ot">&lt;-</span> </span>
<span id="cb2-52"><a href="#cb2-52" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mvrnorm</span>(<span class="at">n =</span> sample_size, </span>
<span id="cb2-53"><a href="#cb2-53" aria-hidden="true" tabindex="-1"></a>                <span class="at">mu =</span> <span class="fu">rep</span>(<span class="dv">0</span>,meta_data[i,<span class="st">"group_n"</span>]),</span>
<span id="cb2-54"><a href="#cb2-54" aria-hidden="true" tabindex="-1"></a>                <span class="at">Sigma =</span> covariance_matrix[group_range, group_range])</span>
<span id="cb2-55"><a href="#cb2-55" aria-hidden="true" tabindex="-1"></a>    data<span class="ot">=</span><span class="fu">as.data.frame</span>(data)</span>
<span id="cb2-56"><a href="#cb2-56" aria-hidden="true" tabindex="-1"></a>    beta_coefficients[meta_data[i, <span class="st">"first_index"</span>]<span class="sc">:</span>meta_data[i, <span class="st">"last_index"</span>]] <span class="ot">&lt;-</span></span>
<span id="cb2-57"><a href="#cb2-57" aria-hidden="true" tabindex="-1"></a>        beta_coefficients[meta_data[i, <span class="st">"first_index"</span>]<span class="sc">:</span>meta_data[i, <span class="st">"last_index"</span>]]<span class="sc">+</span></span>
<span id="cb2-58"><a href="#cb2-58" aria-hidden="true" tabindex="-1"></a>        meta_data[i,<span class="st">"group_effect"</span>]</span>
<span id="cb2-59"><a href="#cb2-59" aria-hidden="true" tabindex="-1"></a>    predictor_names<span class="ot">&lt;-</span><span class="fu">paste0</span>(meta_data[i,<span class="st">"group_name"</span>],<span class="st">"_"</span>,<span class="dv">1</span><span class="sc">:</span>meta_data[i,<span class="st">"group_n"</span>])</span>
<span id="cb2-60"><a href="#cb2-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(beta_coefficients)[meta_data[i, <span class="st">"first_index"</span>]<span class="sc">:</span>meta_data[i, <span class="st">"last_index"</span>]] <span class="ot">&lt;-</span> predictor_names</span>
<span id="cb2-61"><a href="#cb2-61" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(data)[meta_data[i, <span class="st">"first_index"</span>]<span class="sc">:</span>meta_data[i, <span class="st">"last_index"</span>]]<span class="ot">&lt;-</span>predictor_names</span>
<span id="cb2-62"><a href="#cb2-62" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb2-63"><a href="#cb2-63" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-64"><a href="#cb2-64" aria-hidden="true" tabindex="-1"></a>score<span class="ot">=</span><span class="fu">as.matrix</span>(data)<span class="sc">%*%</span>beta_coefficients <span class="co"># score of each sample</span></span>
<span id="cb2-65"><a href="#cb2-65" aria-hidden="true" tabindex="-1"></a><span class="co"># logistic function to get a probability, intercept = 0, </span></span>
<span id="cb2-66"><a href="#cb2-66" aria-hidden="true" tabindex="-1"></a><span class="co"># to decrease prevalence, set p-0.2, negative probabilities into 0</span></span>
<span id="cb2-67"><a href="#cb2-67" aria-hidden="true" tabindex="-1"></a>probabilities <span class="ot">&lt;-</span> ((<span class="dv">1</span><span class="sc">/</span>(<span class="dv">1</span><span class="sc">+</span><span class="fu">exp</span>(<span class="sc">-</span>(<span class="dv">0</span><span class="sc">+</span>score))))<span class="sc">-</span><span class="fu">rnorm</span>(sample_size,<span class="at">m=</span><span class="fl">0.2</span>,<span class="at">sd=</span><span class="fl">0.05</span>))<span class="sc">%&gt;%</span></span>
<span id="cb2-68"><a href="#cb2-68" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ifelse</span>(.<span class="sc">&gt;</span><span class="dv">1</span>,<span class="dv">1</span>,.)<span class="sc">%&gt;%</span><span class="fu">abs</span>()</span>
<span id="cb2-69"><a href="#cb2-69" aria-hidden="true" tabindex="-1"></a>response <span class="ot">&lt;-</span> <span class="fu">rbinom</span>(sample_size, <span class="dv">1</span>, probabilities) </span>
<span id="cb2-70"><a href="#cb2-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-71"><a href="#cb2-71" aria-hidden="true" tabindex="-1"></a>age_distribution<span class="ot">=</span><span class="fu">rchisq</span>(sample_size,<span class="at">df=</span><span class="dv">9</span>)</span>
<span id="cb2-72"><a href="#cb2-72" aria-hidden="true" tabindex="-1"></a>sex_distribution<span class="ot">=</span><span class="fu">sample</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),sample_size,<span class="at">replace=</span><span class="cn">TRUE</span>,<span class="at">prob =</span> <span class="fu">c</span>(<span class="fl">0.45</span>,<span class="fl">0.55</span>))</span>
<span id="cb2-73"><a href="#cb2-73" aria-hidden="true" tabindex="-1"></a>country_distribution<span class="ot">=</span><span class="fu">sample</span>(<span class="fu">c</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">3</span>),sample_size,<span class="at">replace=</span><span class="cn">TRUE</span>,<span class="at">prob =</span> <span class="fu">c</span>(<span class="fl">0.3</span>,<span class="fl">0.2</span>,<span class="fl">0.2</span>,<span class="fl">0.3</span>))</span>
<span id="cb2-74"><a href="#cb2-74" aria-hidden="true" tabindex="-1"></a>treatment_distribution<span class="ot">=</span><span class="fu">sample</span>(<span class="fu">c</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">2</span>),sample_size,<span class="at">replace=</span><span class="cn">TRUE</span>,<span class="at">prob =</span> <span class="fu">c</span>(<span class="fl">0.7</span>,<span class="fl">0.2</span>,<span class="fl">0.1</span>))</span>
<span id="cb2-75"><a href="#cb2-75" aria-hidden="true" tabindex="-1"></a>genotype_distribution<span class="ot">=</span><span class="fu">sample</span>(<span class="fu">c</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">5</span>),sample_size,<span class="at">replace=</span><span class="cn">TRUE</span>,</span>
<span id="cb2-76"><a href="#cb2-76" aria-hidden="true" tabindex="-1"></a>                             <span class="at">prob =</span> <span class="fu">c</span>(<span class="fl">0.08</span>,<span class="fl">0.15</span>,<span class="fl">0.086</span>,<span class="fl">0.3</span>,<span class="fl">0.25</span>,<span class="fl">0.134</span>)) <span class="co">#just random numbers</span></span>
<span id="cb2-77"><a href="#cb2-77" aria-hidden="true" tabindex="-1"></a>phenotype_data<span class="ot">&lt;-</span></span>
<span id="cb2-78"><a href="#cb2-78" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(</span>
<span id="cb2-79"><a href="#cb2-79" aria-hidden="true" tabindex="-1"></a>        <span class="at">id=</span><span class="dv">1</span><span class="sc">:</span>sample_size,</span>
<span id="cb2-80"><a href="#cb2-80" aria-hidden="true" tabindex="-1"></a>        <span class="at">outcome=</span>response,</span>
<span id="cb2-81"><a href="#cb2-81" aria-hidden="true" tabindex="-1"></a>        <span class="at">probabilities=</span>probabilities,</span>
<span id="cb2-82"><a href="#cb2-82" aria-hidden="true" tabindex="-1"></a>        <span class="at">age=</span><span class="fu">ifelse</span>(probabilities<span class="sc">&lt;</span><span class="fl">0.15</span>,age_distribution<span class="dv">-4</span>,</span>
<span id="cb2-83"><a href="#cb2-83" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">ifelse</span>(probabilities<span class="sc">&lt;</span><span class="fl">0.3</span>,age_distribution<span class="dv">-2</span>,</span>
<span id="cb2-84"><a href="#cb2-84" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">ifelse</span>(probabilities<span class="sc">&gt;</span><span class="fl">0.5</span>,age_distribution<span class="sc">+</span><span class="dv">3</span>,</span>
<span id="cb2-85"><a href="#cb2-85" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">ifelse</span>(probabilities<span class="sc">&gt;</span><span class="fl">0.7</span>,age_distribution<span class="sc">+</span><span class="dv">6</span>,age_distribution)))))<span class="sc">%&gt;%</span></span>
<span id="cb2-86"><a href="#cb2-86" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">age=</span><span class="fu">sapply</span>(age,</span>
<span id="cb2-87"><a href="#cb2-87" aria-hidden="true" tabindex="-1"></a>                   <span class="cf">function</span>(x)(x<span class="sc">-</span><span class="fu">min</span>(age))<span class="sc">/</span>(<span class="fu">max</span>(age)<span class="sc">-</span><span class="fu">min</span>(age))<span class="sc">*</span>(<span class="dv">105-65</span>)<span class="sc">+</span><span class="dv">65</span>)<span class="sc">%&gt;%</span><span class="fu">round</span>(<span class="dv">0</span>),</span>
<span id="cb2-88"><a href="#cb2-88" aria-hidden="true" tabindex="-1"></a>           <span class="at">sex=</span>sex_distribution,</span>
<span id="cb2-89"><a href="#cb2-89" aria-hidden="true" tabindex="-1"></a>           <span class="at">country=</span>country_distribution,</span>
<span id="cb2-90"><a href="#cb2-90" aria-hidden="true" tabindex="-1"></a>           <span class="at">treatment=</span>treatment_distribution,</span>
<span id="cb2-91"><a href="#cb2-91" aria-hidden="true" tabindex="-1"></a>           <span class="at">treatment=</span><span class="fu">ifelse</span>(probabilities<span class="sc">&gt;</span><span class="fl">0.7</span>,<span class="dv">1</span>,</span>
<span id="cb2-92"><a href="#cb2-92" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">ifelse</span>(probabilities<span class="sc">&gt;</span><span class="fl">0.8</span>,<span class="dv">2</span>,treatment)),</span>
<span id="cb2-93"><a href="#cb2-93" aria-hidden="true" tabindex="-1"></a>           <span class="at">genotype=</span>genotype_distribution,</span>
<span id="cb2-94"><a href="#cb2-94" aria-hidden="true" tabindex="-1"></a>           <span class="at">genotype=</span><span class="fu">ifelse</span>(probabilities<span class="sc">&lt;</span><span class="fl">0.1</span>,<span class="dv">0</span>,</span>
<span id="cb2-95"><a href="#cb2-95" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">ifelse</span>(probabilities<span class="sc">&lt;</span><span class="fl">0.15</span>,<span class="dv">1</span>,</span>
<span id="cb2-96"><a href="#cb2-96" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">ifelse</span>(probabilities<span class="sc">&gt;</span><span class="fl">0.7</span>,<span class="dv">3</span>,</span>
<span id="cb2-97"><a href="#cb2-97" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">ifelse</span>(probabilities<span class="sc">&gt;</span><span class="fl">0.8</span>,<span class="dv">4</span>,</span>
<span id="cb2-98"><a href="#cb2-98" aria-hidden="true" tabindex="-1"></a>                                                <span class="fu">ifelse</span>(probabilities<span class="sc">&gt;</span><span class="fl">0.9</span>,<span class="dv">5</span>,genotype))))),</span>
<span id="cb2-99"><a href="#cb2-99" aria-hidden="true" tabindex="-1"></a>           <span class="at">age=</span><span class="fu">ifelse</span>(outcome<span class="sc">==</span><span class="dv">0</span><span class="sc">&amp;</span>genotype<span class="sc">==</span><span class="dv">0</span>,age<span class="sc">+</span><span class="dv">10</span>,</span>
<span id="cb2-100"><a href="#cb2-100" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">ifelse</span>(outcome<span class="sc">==</span><span class="dv">0</span><span class="sc">&amp;</span>genotype<span class="sc">==</span><span class="dv">1</span>,age<span class="sc">+</span><span class="dv">5</span>,</span>
<span id="cb2-101"><a href="#cb2-101" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">ifelse</span>(genotype<span class="sc">==</span><span class="dv">4</span>,age<span class="dv">-3</span>,</span>
<span id="cb2-102"><a href="#cb2-102" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">ifelse</span>(genotype<span class="sc">==</span><span class="dv">5</span>,age<span class="dv">-2</span>,age)))),</span>
<span id="cb2-103"><a href="#cb2-103" aria-hidden="true" tabindex="-1"></a>           <span class="at">genotype=</span><span class="fu">ifelse</span>(outcome<span class="sc">==</span><span class="dv">1</span><span class="sc">&amp;</span>genotype<span class="sc">==</span><span class="dv">3</span><span class="sc">&amp;</span>age<span class="sc">&gt;</span><span class="dv">83</span><span class="sc">&amp;</span>age<span class="sc">&lt;</span><span class="dv">92</span>,<span class="dv">4</span>,genotype),</span>
<span id="cb2-104"><a href="#cb2-104" aria-hidden="true" tabindex="-1"></a>           <span class="at">genotype=</span><span class="fu">ifelse</span>(outcome<span class="sc">==</span><span class="dv">1</span><span class="sc">&amp;</span>genotype<span class="sc">==</span><span class="dv">4</span><span class="sc">&amp;</span>age<span class="sc">&gt;</span><span class="dv">87</span>,<span class="dv">5</span>,genotype))</span>
<span id="cb2-105"><a href="#cb2-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-106"><a href="#cb2-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-107"><a href="#cb2-107" aria-hidden="true" tabindex="-1"></a>all_data<span class="ot">=</span><span class="fu">inner_join</span>(phenotype_data,data<span class="sc">%&gt;%</span><span class="fu">mutate</span>(<span class="at">id=</span><span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>()),<span class="at">by=</span><span class="st">"id"</span>)</span>
<span id="cb2-108"><a href="#cb2-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-109"><a href="#cb2-109" aria-hidden="true" tabindex="-1"></a><span class="co">#write_rds(all_data,"./docs/projects/data/llfs_simulated_data.rds")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="load-data" class="level3" data-number="2.3">
<h3 data-number="2.3" class="anchored" data-anchor-id="load-data"><span class="header-section-number">2.3</span> Load Data</h3>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>datapath<span class="ot">&lt;-</span><span class="st">"C:/Users/kmkim/Desktop/my_project/website/docs/projects/data/llfs_simulated_data.rds"</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="co">#datapath&lt;-"C:/Users/kkm/Desktop/projects/website/docs/projects/data/llfs_simulated_data.rds"</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a> simulated_data<span class="ot">=</span><span class="fu">read_rds</span>(datapath)<span class="sc">%&gt;%</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>     dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="sc">-</span>probabilities)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a> all_data<span class="ot">=</span>simulated_data<span class="sc">%&gt;%</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a> <span class="fu">mutate</span>(</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>      <span class="at">outcome=</span><span class="fu">ifelse</span>(outcome<span class="sc">==</span><span class="dv">0</span>,<span class="st">"negative"</span>,<span class="st">"positive"</span>),</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">outcome=</span><span class="fu">factor</span>(outcome,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">"negative"</span>,<span class="st">"positive"</span>)),</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">sex=</span><span class="fu">ifelse</span>(sex<span class="sc">==</span><span class="dv">0</span>,<span class="st">"man"</span>,<span class="st">"woman"</span>),</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>      <span class="at">sex=</span><span class="fu">factor</span>(sex,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">"man"</span>,<span class="st">"woman"</span>)),</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>      <span class="at">country=</span><span class="fu">ifelse</span>(country<span class="sc">==</span><span class="dv">0</span>,<span class="st">"center_a"</span>,<span class="fu">ifelse</span>(country<span class="sc">==</span><span class="dv">1</span>,<span class="st">"center_b"</span>,<span class="fu">ifelse</span>(country<span class="sc">==</span><span class="dv">2</span>,<span class="st">"center_c"</span>,<span class="st">"center_d"</span>))),</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>      <span class="at">country=</span><span class="fu">factor</span>(country,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">"center_a"</span>,<span class="st">"center_b"</span>,<span class="st">"center_c"</span>,<span class="st">"center_d"</span>)),</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>      <span class="at">treatment=</span><span class="fu">ifelse</span>(treatment<span class="sc">==</span><span class="dv">0</span>,<span class="st">"trmnt1"</span>,<span class="fu">ifelse</span>(treatment<span class="sc">==</span><span class="dv">1</span>,<span class="st">"trmnt2"</span>,<span class="st">"trmnt3"</span>)),</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>      <span class="at">treatment=</span><span class="fu">factor</span>(treatment,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">"trmnt1"</span>,<span class="st">"trmnt2"</span>,<span class="st">"trmnt3"</span>)),</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>      <span class="at">genotype=</span><span class="fu">ifelse</span>(genotype<span class="sc">==</span><span class="dv">0</span>,<span class="st">"e2/e2"</span>,</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>      <span class="fu">ifelse</span>(genotype<span class="sc">==</span><span class="dv">1</span>,<span class="st">"e2/e3"</span>,</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>      <span class="fu">ifelse</span>(genotype<span class="sc">==</span><span class="dv">2</span>,<span class="st">"e2/e4"</span>,</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>      <span class="fu">ifelse</span>(genotype<span class="sc">==</span><span class="dv">3</span>,<span class="st">"e3/e3"</span>,</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>      <span class="fu">ifelse</span>(genotype<span class="sc">==</span><span class="dv">4</span>,<span class="st">"e3/e4"</span>,<span class="st">"e4/e4"</span>))))),</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>      <span class="at">genotype=</span><span class="fu">factor</span>(genotype,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">"e2/e2"</span>,<span class="st">"e2/e3"</span>,<span class="st">"e2/e4"</span>,<span class="st">"e3/e3"</span>,<span class="st">"e3/e4"</span>,<span class="st">"e4/e4"</span>))</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a> <span class="fu">names</span>(all_data)[<span class="dv">7</span><span class="sc">:</span><span class="fu">ncol</span>(all_data)]<span class="ot">&lt;-</span><span class="fu">paste0</span>(<span class="st">"meta"</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">10000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">

</div>
</section>
</section>
<section id="data-description" class="level2" data-number="3">
<h2 data-number="3" class="anchored" data-anchor-id="data-description"><span class="header-section-number">3</span> Data Description</h2>
<p>This data include 1000 samples and 10006 variables:</p>
<ul>
<li>outcome: a disease status (positive, negative), positive is an affected status, negative is an unaffected status, and the reference group is negative.</li>
<li>age: an age</li>
<li>sex: a sex status (woman, man) and the reference group is man.</li>
<li>country: a country where data were collected (center_d, center_b, center_a, center_c) and the reference group is center_a.</li>
<li>treatment: a treatment for the disease (trmnt2, trmnt1, trmnt3) and the reference group is trmnt1.</li>
<li>genotype: a genotype of APOE.
<ul>
<li>the apolipoprotein <span class="math inline">\(\epsilon\)</span> (APOE) is a protein produced in the metabolic pathway of fats in mammals, a genotype of which seems to be related to Alzheimer’s disease (AD). APOE is polymorphic and has three major alleles, <span class="math inline">\(\epsilon 2\)</span> (e2), <span class="math inline">\(\epsilon 3\)</span>(e3), and <span class="math inline">\(\epsilon 4\)</span> (e4). The statistics of the polymorphism are 8.4% for e2, 77.9% for e3, and 13.7% for e4 in worldwide allel frequency, respectively. It is known that the e2, e3, and e4 allels are associated with the protective factor, the neutral one, and the risk one with regard to AD. However, this finding has not been replicated in a large population. Therefore, it is known that we do not know their true associations with AD in the true population. <a href="https://en.wikipedia.org/wiki/Apolipoprotein_E">(from Wiki)</a></li>
<li>There are 6 combinations of the genotypes:
<ul>
<li>e2/e2</li>
<li>e2/e3</li>
<li>e2/e4</li>
<li>e3/e3 (reference group)</li>
<li>e3/e4</li>
<li>e4/e4</li>
</ul></li>
</ul></li>
<li>meta1 ~ meta10000: a list of metabolites that were blood-sampled from the APOE carriers.</li>
</ul>
</section>
</div>



</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>