<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.543">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Kwangmin Kim">
<meta name="dcterms.date" content="2023-01-03">
<meta name="description" content="Due to security concerns, it is difficult to display the data used in this project. This project was not only a project I worked at Columbia University Irving Medical Center, but also used for my graduate school graduation research practicum. Although the goal of each step was different, the subject matter is the same, so I will combine them to explain the project. 보안 문제로 인해 이 프로젝트에서 사용된 데이터를 공유하기 어렵습니다. 이 프로젝트는 Columbia University Irving Medical Center에서 근무했던 프로젝트 였던 것 뿐만 아니라 제 대학원 졸업 연구 발표용으로도 사용됐습니다. 각 단계에서의 목표는 달랐지만 주제는 같으므로 통합해서 설명하겠습니다.">

<title>Kwangmin Kim - Description</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../.././images/logo.png" alt="" class="navbar-logo">
    </a>
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">Kwangmin Kim</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../docs/CV/index.html"> 
<span class="menu-text">CV</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="../../../docs/projects/index.html" aria-current="page"> 
<span class="menu-text">Projects</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../about.html"> 
<span class="menu-text">Me</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/kmink3225"> <i class="bi bi-github" role="img" aria-label="Github">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://www.linkedin.com/in/kwangmin-kim-a5241b200/"> <i class="bi bi-linkedin" role="img" aria-label="Linkedin">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../docs/projects/LLFS/project_description.html">LLFS</a></li><li class="breadcrumb-item"><a href="../../../docs/projects/LLFS/project_description.html">Description</a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../docs/projects/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Projects</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false">
 <span class="menu-text">Baseline Fitting</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../docs/projects/baseline_subtraction/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Baseline Fitting Algorithm Improvement</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false">
 <span class="menu-text">FDA Validation</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../docs/projects/dsp_validation/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">FDA Algorithms Validation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">
 <span class="menu-text">Diagnostic Device QC</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../docs/projects/qc_platform/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DevOps Project: Diagnostic Device QC Platform Construction</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">
 <span class="menu-text">LLFS</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../docs/projects/LLFS/project_description.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Description</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false">
 <span class="menu-text">Heavy Metal Removal</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../docs/projects/heavy_metal/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Project Description</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false">
 <span class="menu-text">Phellinus Linteus Effects</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../docs/projects/phellinus_linteus/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Project Description</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../docs/projects/LLFS/project_description.html">LLFS</a></li><li class="breadcrumb-item"><a href="../../../docs/projects/LLFS/project_description.html">Description</a></li></ol></nav>
<div class="quarto-title">
<h1 class="title">Description</h1>
<p class="subtitle lead">Project Description</p>
</div>

<div>
  <div class="description">
    Due to security concerns, it is difficult to display the data used in this project. This project was not only a project I worked at Columbia University Irving Medical Center, but also used for my graduate school graduation research practicum. Although the goal of each step was different, the subject matter is the same, so I will combine them to explain the project. 보안 문제로 인해 이 프로젝트에서 사용된 데이터를 공유하기 어렵습니다. 이 프로젝트는 Columbia University Irving Medical Center에서 근무했던 프로젝트 였던 것 뿐만 아니라 제 대학원 졸업 연구 발표용으로도 사용됐습니다. 각 단계에서의 목표는 달랐지만 주제는 같으므로 통합해서 설명하겠습니다.
  </div>
</div>


<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Kwangmin Kim </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">January 3, 2023</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<ul class="nav nav-pills" id="language-tab" role="tablist">
<li class="nav-item" role="presentation">
<button class="nav-link active" id="Korean-tab" data-bs-toggle="tab" data-bs-target="#Korean" type="button" role="tab" aria-controls="Korean" aria-selected="true">
Korean
</button>
</li>
<li class="nav-item" role="presentation">
<button class="nav-link" id="English-tab" data-bs-toggle="tab" data-bs-target="#English" type="button" role="tab" aria-controls="knitr" aria-selected="false">
English
</button>
</li>
<div class="tab-content" id="language-tabcontent">

<div id="Korean" class="tab-pane fade show active" role="tabpanel" aria-labelledby="Korean-tab">
<section id="개요" class="level2" data-number="1">
<h2 data-number="1" class="anchored" data-anchor-id="개요"><span class="header-section-number">1</span> 개요</h2>
<ul>
<li>목적: 2차 세계 대전 이후 Baby Boommer Generation의 대규모 치매 발병에 대비한 치매 인자 규명</li>
<li>기간: 2018.12 - 2020.04 (17 months)</li>
<li>역할: Research Assistant and Statistician</li>
<li>참여 인원: 1명의 역학 교수 (mentor), 1명의 생물통계학 교수 (mentor), 2명의 신경외과 의사 (mentor), 2명의 통계학자 (연구원), 1명의 역학자 (연구원), 1명의 생화학자 (연구원)</li>
</ul>
</section>
<section id="background" class="level2" data-number="2">
<h2 data-number="2" class="anchored" data-anchor-id="background"><span class="header-section-number">2</span> Background</h2>
<p>알츠하이머병(Alzheimer’s Disease, AD)은 수백만 명의 미국인에게 영향을 미치는 가장 흔한 형태의 치매이다. 알츠하이머병은 기억력, 사고력 및 행동에 영향을 주지만 증상이 나타나기까지 거의 20년에 걸쳐 진행이 된다. 따라서 전 임상 단계 (Pre-clinical Stage)에서 생리학을 이해하는 것이 필수적이다. 유전적 요인이 AD에 거의 50% 기여하는 것으로 추정된다. 유전자가 세포 환경을 변경하여 알츠하이머병 위험에 어떻게 기여하는지 더 잘 이해하기 위해 AD와 연관이 있는 유전자인 APOE를 보유한 사람들의 대사체(Metabolome)를 조사했다. 대사체는 유전체(Genome)과 단백질체(Proteome)에서 생성된 산물을 의미한다. 이러한 생화학 부산물은 유전적 요인과 환경적 요인 모두의 영향을 받는다. 모집단은 장수마을에 사는 Caucasian (백인) 참여자들이다.</p>
</section>
<section id="objective" class="level2" data-number="3">
<h2 data-number="3" class="anchored" data-anchor-id="objective"><span class="header-section-number">3</span> Objective</h2>
<p>LLFS(Long Life Family Study) 프로젝트의 목적은 유전체, 전사체, 단백질체 및 대사체 단계를 통해 유전체에서 대사체 단계에 이르는 여러 단계에서 통계 및 기계 학습을 사용하여 분석 파이프라인을 구축하고 알츠하이머병에 대한 중요한 바이오마커를 식별하는 것이다.</p>
</section>
<section id="analysis-pipeline" class="level2" data-number="4">
<h2 data-number="4" class="anchored" data-anchor-id="analysis-pipeline"><span class="header-section-number">4</span> Analysis Pipeline</h2>
<section id="overview" class="level3" data-number="4.1">
<h3 data-number="4.1" class="anchored" data-anchor-id="overview"><span class="header-section-number">4.1</span> Overview</h3>
<div class="cell" data-layout-align="default">
<div class="cell-output-display">
<div>
<p></p><figure class="figure"><p></p>
<div>
<svg width="672" height="480" viewbox="0.00 0.00 778.83 425.00" xmlns="http://www.w3.org/2000/svg" xlink="http://www.w3.org/1999/xlink" style="; max-width: none; max-height: none">
<g id="graph0" class="graph" transform="scale(1 1) rotate(0) translate(4 421)">
<title>G</title>
<polygon fill="white" stroke="transparent" points="-4,4 -4,-421 774.83,-421 774.83,4 -4,4"></polygon>
<g id="clust1" class="cluster">
<title>cluster0</title>
<polygon fill="none" stroke="black" points="8,-224 8,-409 230.54,-409 230.54,-224 8,-224"></polygon>
<text text-anchor="middle" x="119.27" y="-392.4" font-family="Times,serif" font-size="14.00">Data Collection</text>
</g>
<g id="clust2" class="cluster">
<title>cluster1</title>
<polygon fill="none" stroke="black" points="250.54,-170 250.54,-301 424.86,-301 424.86,-170 250.54,-170"></polygon>
<text text-anchor="middle" x="337.7" y="-284.4" font-family="Times,serif" font-size="14.00">Quality Control</text>
</g>
<g id="clust3" class="cluster">
<title>cluster2</title>
<polygon fill="none" stroke="black" points="444.86,-8 444.86,-247 588.08,-247 588.08,-8 444.86,-8"></polygon>
<text text-anchor="middle" x="516.47" y="-230.4" font-family="Times,serif" font-size="14.00">Analytics</text>
</g>
<g id="clust4" class="cluster">
<title>cluster3</title>
<polygon fill="none" stroke="black" points="608.08,-8 608.08,-85 770.83,-85 770.83,-8 608.08,-8"></polygon>
<text text-anchor="middle" x="689.46" y="-68.4" font-family="Times,serif" font-size="14.00">Reporting and Conclusion</text>
</g>
<!-- Multi_Centered_Blood_Sampling -->
<g id="node1" class="node">
<title>Multi_Centered_Blood_Sampling</title>
<polygon fill="none" stroke="black" points="222.32,-376 16.23,-376 16.23,-340 222.32,-340 222.32,-376"></polygon>
<text text-anchor="middle" x="119.27" y="-353.8" font-family="Times,serif" font-size="14.00">Multi_Centered_Blood_Sampling</text>
</g>
<!-- Mass_Spectrometry -->
<g id="node2" class="node">
<title>Mass_Spectrometry</title>
<polygon fill="none" stroke="black" points="183.25,-322 55.29,-322 55.29,-286 183.25,-286 183.25,-322"></polygon>
<text text-anchor="middle" x="119.27" y="-299.8" font-family="Times,serif" font-size="14.00">Mass_Spectrometry</text>
</g>
<!-- Data_Transfer -->
<g id="node3" class="node">
<title>Data_Transfer</title>
<polygon fill="none" stroke="black" points="167.61,-268 70.93,-268 70.93,-232 167.61,-232 167.61,-268"></polygon>
<text text-anchor="middle" x="119.27" y="-245.8" font-family="Times,serif" font-size="14.00">Data_Transfer</text>
</g>
<!-- Identify_Anomaly_Data -->
<g id="node4" class="node">
<title>Identify_Anomaly_Data</title>
<polygon fill="none" stroke="black" points="414.05,-268 261.35,-268 261.35,-232 414.05,-232 414.05,-268"></polygon>
<text text-anchor="middle" x="337.7" y="-245.8" font-family="Times,serif" font-size="14.00">Identify_Anomaly_Data</text>
</g>
<!-- Data_Transfer&#45;&gt;Identify_Anomaly_Data -->
<g id="edge1" class="edge">
<title>Data_Transfer-&gt;Identify_Anomaly_Data</title>
<path fill="none" stroke="black" d="M167.77,-250C189.1,-250 215.09,-250 240.22,-250"></path>
<polygon fill="black" stroke="black" points="240.55,-253.5 250.55,-250 240.55,-246.5 240.55,-253.5"></polygon>
</g>
<!-- Identify_Missing_Values -->
<g id="node5" class="node">
<title>Identify_Missing_Values</title>
<polygon fill="none" stroke="black" points="417.02,-214 258.39,-214 258.39,-178 417.02,-178 417.02,-214"></polygon>
<text text-anchor="middle" x="337.7" y="-191.8" font-family="Times,serif" font-size="14.00">Identify_Missing_Values</text>
</g>
<!-- EDA -->
<g id="node6" class="node">
<title>EDA</title>
<polygon fill="none" stroke="black" points="543.47,-214 489.47,-214 489.47,-178 543.47,-178 543.47,-214"></polygon>
<text text-anchor="middle" x="516.47" y="-191.8" font-family="Times,serif" font-size="14.00">EDA</text>
</g>
<!-- Identify_Missing_Values&#45;&gt;EDA -->
<g id="edge2" class="edge">
<title>Identify_Missing_Values-&gt;EDA</title>
<path fill="none" stroke="black" d="M417.01,-196C422.8,-196 428.66,-196 434.47,-196"></path>
<polygon fill="black" stroke="black" points="434.87,-199.5 444.87,-196 434.87,-192.5 434.87,-199.5"></polygon>
</g>
<!-- Data_Mining -->
<g id="node7" class="node">
<title>Data_Mining</title>
<polygon fill="none" stroke="black" points="561.63,-160 471.31,-160 471.31,-124 561.63,-124 561.63,-160"></polygon>
<text text-anchor="middle" x="516.47" y="-137.8" font-family="Times,serif" font-size="14.00">Data_Mining</text>
</g>
<!-- Statistical_Analysis -->
<g id="node8" class="node">
<title>Statistical_Analysis</title>
<polygon fill="none" stroke="black" points="580.18,-106 452.76,-106 452.76,-70 580.18,-70 580.18,-106"></polygon>
<text text-anchor="middle" x="516.47" y="-83.8" font-family="Times,serif" font-size="14.00">Statistical_Analysis</text>
</g>
<!-- Machine_Learning -->
<g id="node9" class="node">
<title>Machine_Learning</title>
<polygon fill="none" stroke="black" points="577.48,-52 455.46,-52 455.46,-16 577.48,-16 577.48,-52"></polygon>
<text text-anchor="middle" x="516.47" y="-29.8" font-family="Times,serif" font-size="14.00">Machine_Learning</text>
</g>
<!-- Share_with_Faculty -->
<g id="node10" class="node">
<title>Share_with_Faculty</title>
<polygon fill="none" stroke="black" points="753.71,-52 625.2,-52 625.2,-16 753.71,-16 753.71,-52"></polygon>
<text text-anchor="middle" x="689.46" y="-29.8" font-family="Times,serif" font-size="14.00">Share_with_Faculty</text>
</g>
<!-- Machine_Learning&#45;&gt;Share_with_Faculty -->
<g id="edge3" class="edge">
<title>Machine_Learning-&gt;Share_with_Faculty</title>
<path fill="none" stroke="black" d="M577.75,-34C584.35,-34 591.16,-34 598.02,-34"></path>
<polygon fill="black" stroke="black" points="598.08,-37.5 608.08,-34 598.08,-30.5 598.08,-37.5"></polygon>
</g>
</g>
</svg>
</div>
<p></p></figure><p></p>
</div>
</div>
</div>
<p>Data는 장수 마을에 거주하는 백인을 대상으로 New York, Bonston, Pittsburgh 및 Denmark에 있는 여러 medical centers에서 sampled blood를 MS Spectromtetry로 Digitalization을 했다. 여러 과정을 통해 data를 csv형태로 받아 data의 QC(Quality Control)를 진행한뒤 Data 분석 업무를 수행했다. EDA (Exploratory Data Analysis) 와 Data Mining을 통해 data에 대한 이해도를 높였고 이를 토대로 통계 분석과 machine learning을 이용하여 이 data에 적합한 모형을 찾았다. 모든 결과물은 The Taub Institute for Research on Alzheimer’s Disease and the Aging Brain 의 biostaticians, medical doctors, biologists, neurologists, bioinformaticians 및 epidemiologists와 공유를 했다.</p>
</section>
<section id="data-qc-quality-control" class="level3" data-number="4.2">
<h3 data-number="4.2" class="anchored" data-anchor-id="data-qc-quality-control"><span class="header-section-number">4.2</span> Data QC (Quality Control)</h3>
<div class="cell" data-fig-height="8" data-layout-align="default">
<div class="cell-output-display">
<div>
<p></p><figure class="figure"><p></p>
<div>
<svg width="672" height="768" viewbox="0.00 0.00 268.00 838.40" xmlns="http://www.w3.org/2000/svg" xlink="http://www.w3.org/1999/xlink" style="; max-width: none; max-height: none">
<g id="graph0" class="graph" transform="scale(1 1) rotate(0) translate(4 834.4)">
<title>G</title>
<polygon fill="white" stroke="transparent" points="-4,4 -4,-834.4 264,-834.4 264,4 -4,4"></polygon>
<g id="clust3" class="cluster">
<title>cluster2</title>
<polygon fill="none" stroke="black" points="54,-8 54,-228.8 206,-228.8 206,-8 54,-8"></polygon>
<text text-anchor="middle" x="130" y="-212.2" font-family="Times,serif" font-size="14.00">Data Preprocessing</text>
</g>
<g id="clust1" class="cluster">
<title>cluster0</title>
<polygon fill="none" stroke="black" points="8,-236.8 8,-822.4 252,-822.4 252,-236.8 8,-236.8"></polygon>
<text text-anchor="middle" x="130" y="-805.8" font-family="Times,serif" font-size="14.00">Data Quality Control</text>
</g>
<g id="clust2" class="cluster">
<title>cluster1</title>
<polygon fill="none" stroke="black" points="18,-452.8 18,-673.6 242,-673.6 242,-452.8 18,-452.8"></polygon>
<text text-anchor="middle" x="130" y="-657" font-family="Times,serif" font-size="14.00">Missing Value Analysis</text>
</g>
<!-- identify_anomaly_data -->
<g id="node1" class="node">
<title>identify_anomaly_data</title>
<polygon fill="none" stroke="black" points="202.57,-789.6 57.43,-789.6 57.43,-753.6 202.57,-753.6 202.57,-789.6"></polygon>
<text text-anchor="middle" x="130" y="-767.4" font-family="Times,serif" font-size="14.00">identify_anomaly_data</text>
</g>
<!-- identify_missing_values -->
<g id="node2" class="node">
<title>identify_missing_values</title>
<polygon fill="none" stroke="black" points="206.38,-717.6 53.62,-717.6 53.62,-681.6 206.38,-681.6 206.38,-717.6"></polygon>
<text text-anchor="middle" x="130" y="-695.4" font-family="Times,serif" font-size="14.00">identify_missing_values</text>
</g>
<!-- identify_anomaly_data&#45;&gt;identify_missing_values -->
<g id="edge3" class="edge">
<title>identify_anomaly_data-&gt;identify_missing_values</title>
<path fill="none" stroke="black" d="M130,-753.3C130,-745.58 130,-736.31 130,-727.71"></path>
<polygon fill="black" stroke="black" points="133.5,-727.7 130,-717.7 126.5,-727.7 133.5,-727.7"></polygon>
</g>
<!-- Missing_Completely_At_Random -->
<g id="node3" class="node">
<title>Missing_Completely_At_Random</title>
<polygon fill="none" stroke="black" points="234.39,-640.8 25.61,-640.8 25.61,-604.8 234.39,-604.8 234.39,-640.8"></polygon>
<text text-anchor="middle" x="130" y="-618.6" font-family="Times,serif" font-size="14.00">Missing_Completely_At_Random</text>
</g>
<!-- identify_missing_values&#45;&gt;Missing_Completely_At_Random -->
<g id="edge4" class="edge">
<title>identify_missing_values-&gt;Missing_Completely_At_Random</title>
<path fill="none" stroke="black" d="M130,-681.25C130,-681.1 130,-680.95 130,-680.8"></path>
<polygon fill="black" stroke="black" points="133.5,-683.6 130,-673.6 126.5,-683.6 133.5,-683.6"></polygon>
</g>
<!-- Missing_At_Random -->
<g id="node4" class="node">
<title>Missing_At_Random</title>
<polygon fill="none" stroke="black" points="198.07,-568.8 61.93,-568.8 61.93,-532.8 198.07,-532.8 198.07,-568.8"></polygon>
<text text-anchor="middle" x="130" y="-546.6" font-family="Times,serif" font-size="14.00">Missing_At_Random</text>
</g>
<!-- Missing_Completely_At_Random&#45;&gt;Missing_At_Random -->
<g id="edge1" class="edge">
<title>Missing_Completely_At_Random-&gt;Missing_At_Random</title>
<path fill="none" stroke="black" d="M130,-604.5C130,-596.78 130,-587.51 130,-578.91"></path>
<polygon fill="black" stroke="black" points="133.5,-578.9 130,-568.9 126.5,-578.9 133.5,-578.9"></polygon>
</g>
<!-- Missing_Not_at_Random -->
<g id="node5" class="node">
<title>Missing_Not_at_Random</title>
<polygon fill="none" stroke="black" points="210.17,-496.8 49.83,-496.8 49.83,-460.8 210.17,-460.8 210.17,-496.8"></polygon>
<text text-anchor="middle" x="130" y="-474.6" font-family="Times,serif" font-size="14.00">Missing_Not_at_Random</text>
</g>
<!-- Missing_At_Random&#45;&gt;Missing_Not_at_Random -->
<g id="edge2" class="edge">
<title>Missing_At_Random-&gt;Missing_Not_at_Random</title>
<path fill="none" stroke="black" d="M130,-532.5C130,-524.78 130,-515.51 130,-506.91"></path>
<polygon fill="black" stroke="black" points="133.5,-506.9 130,-496.9 126.5,-506.9 133.5,-506.9"></polygon>
</g>
<!-- either_imputation_or_omission -->
<g id="node6" class="node">
<title>either_imputation_or_omission</title>
<polygon fill="none" stroke="black" points="225.76,-424.8 34.24,-424.8 34.24,-388.8 225.76,-388.8 225.76,-424.8"></polygon>
<text text-anchor="middle" x="130" y="-402.6" font-family="Times,serif" font-size="14.00">either_imputation_or_omission</text>
</g>
<!-- Missing_Not_at_Random&#45;&gt;either_imputation_or_omission -->
<g id="edge5" class="edge">
<title>Missing_Not_at_Random-&gt;either_imputation_or_omission</title>
<path fill="none" stroke="black" d="M130,-460.5C130,-452.78 130,-443.51 130,-434.91"></path>
<polygon fill="black" stroke="black" points="133.5,-434.9 130,-424.9 126.5,-434.9 133.5,-434.9"></polygon>
</g>
<!-- communication_with_labs -->
<g id="node7" class="node">
<title>communication_with_labs</title>
<polygon fill="none" stroke="black" points="212.03,-352.8 47.97,-352.8 47.97,-316.8 212.03,-316.8 212.03,-352.8"></polygon>
<text text-anchor="middle" x="130" y="-330.6" font-family="Times,serif" font-size="14.00">communication_with_labs</text>
</g>
<!-- either_imputation_or_omission&#45;&gt;communication_with_labs -->
<g id="edge6" class="edge">
<title>either_imputation_or_omission-&gt;communication_with_labs</title>
<path fill="none" stroke="black" d="M130,-388.5C130,-380.78 130,-371.51 130,-362.91"></path>
<polygon fill="black" stroke="black" points="133.5,-362.9 130,-352.9 126.5,-362.9 133.5,-362.9"></polygon>
</g>
<!-- set_data_inclusion_exclusion_criteria -->
<g id="node8" class="node">
<title>set_data_inclusion_exclusion_criteria</title>
<polygon fill="none" stroke="black" points="243.99,-280.8 16.01,-280.8 16.01,-244.8 243.99,-244.8 243.99,-280.8"></polygon>
<text text-anchor="middle" x="130" y="-258.6" font-family="Times,serif" font-size="14.00">set_data_inclusion_exclusion_criteria</text>
</g>
<!-- communication_with_labs&#45;&gt;set_data_inclusion_exclusion_criteria -->
<g id="edge7" class="edge">
<title>communication_with_labs-&gt;set_data_inclusion_exclusion_criteria</title>
<path fill="none" stroke="black" d="M130,-316.5C130,-308.78 130,-299.51 130,-290.91"></path>
<polygon fill="black" stroke="black" points="133.5,-290.9 130,-280.9 126.5,-290.9 133.5,-290.9"></polygon>
</g>
<!-- Data_Transformation -->
<g id="node9" class="node">
<title>Data_Transformation</title>
<polygon fill="none" stroke="black" points="198.01,-196 61.99,-196 61.99,-160 198.01,-160 198.01,-196"></polygon>
<text text-anchor="middle" x="130" y="-173.8" font-family="Times,serif" font-size="14.00">Data_Transformation</text>
</g>
<!-- set_data_inclusion_exclusion_criteria&#45;&gt;Data_Transformation -->
<g id="edge10" class="edge">
<title>set_data_inclusion_exclusion_criteria-&gt;Data_Transformation</title>
<path fill="none" stroke="black" d="M130,-244.64C130,-242.86 130,-240.97 130,-239.02"></path>
<polygon fill="black" stroke="black" points="133.5,-238.8 130,-228.8 126.5,-238.8 133.5,-238.8"></polygon>
</g>
<!-- Log_Transformation -->
<g id="node10" class="node">
<title>Log_Transformation</title>
<polygon fill="none" stroke="black" points="196.14,-124 63.86,-124 63.86,-88 196.14,-88 196.14,-124"></polygon>
<text text-anchor="middle" x="130" y="-101.8" font-family="Times,serif" font-size="14.00">Log_Transformation</text>
</g>
<!-- Data_Transformation&#45;&gt;Log_Transformation -->
<g id="edge8" class="edge">
<title>Data_Transformation-&gt;Log_Transformation</title>
<path fill="none" stroke="black" d="M130,-159.7C130,-151.98 130,-142.71 130,-134.11"></path>
<polygon fill="black" stroke="black" points="133.5,-134.1 130,-124.1 126.5,-134.1 133.5,-134.1"></polygon>
</g>
<!-- Standardization -->
<g id="node11" class="node">
<title>Standardization</title>
<polygon fill="none" stroke="black" points="181.86,-52 78.14,-52 78.14,-16 181.86,-16 181.86,-52"></polygon>
<text text-anchor="middle" x="130" y="-29.8" font-family="Times,serif" font-size="14.00">Standardization</text>
</g>
<!-- Log_Transformation&#45;&gt;Standardization -->
<g id="edge9" class="edge">
<title>Log_Transformation-&gt;Standardization</title>
<path fill="none" stroke="black" d="M130,-87.7C130,-79.98 130,-70.71 130,-62.11"></path>
<polygon fill="black" stroke="black" points="133.5,-62.1 130,-52.1 126.5,-62.1 133.5,-62.1"></polygon>
</g>
</g>
</svg>
</div>
<p></p></figure><p></p>
</div>
</div>
</div>
<p>Data의 품질 관리를 위해 data를 생성한 biochemists와 소통하여 실험실 기준에 따라 결측치와 이상치를 구분하여 labeling을 수행했고 missing value analysis를 통해 결과에 따라 medical doctors를 포함한 다른 faculty members와 상의하여 결측치 처리를 했다. data QC criteria는 rowwise 와 columwise sum의 합이 sample size에 대하여 missing values의 비율이 5%가 넘는 환자와 변수는 분석 대상에서 제외 됐다. 모든 metabolites data는 log transoformation 과 standardization을 통해 data의 단위를 표준화 했다.</p>
</section>
<section id="methods" class="level3" data-number="4.3">
<h3 data-number="4.3" class="anchored" data-anchor-id="methods"><span class="header-section-number">4.3</span> Methods</h3>
<div class="cell" data-layout-align="default">
<div class="cell-output-display">
<div>
<p></p><figure class="figure"><p></p>
<div>
<svg width="672" height="480" viewbox="0.00 0.00 212.00 316.80" xmlns="http://www.w3.org/2000/svg" xlink="http://www.w3.org/1999/xlink" style="; max-width: none; max-height: none">
<g id="graph0" class="graph" transform="scale(1 1) rotate(0) translate(4 312.8)">
<title>G</title>
<polygon fill="white" stroke="transparent" points="-4,4 -4,-312.8 208,-312.8 208,4 -4,4"></polygon>
<g id="clust1" class="cluster">
<title>cluster2</title>
<polygon fill="none" stroke="black" points="8,-8 8,-300.8 196,-300.8 196,-8 8,-8"></polygon>
<text text-anchor="middle" x="102" y="-284.2" font-family="Times,serif" font-size="14.00">Data Analytics</text>
</g>
<!-- Exploratory_Data_Analysis -->
<g id="node1" class="node">
<title>Exploratory_Data_Analysis</title>
<polygon fill="none" stroke="black" points="188.3,-268 15.7,-268 15.7,-232 188.3,-232 188.3,-268"></polygon>
<text text-anchor="middle" x="102" y="-245.8" font-family="Times,serif" font-size="14.00">Exploratory_Data_Analysis</text>
</g>
<!-- Data_Minig -->
<g id="node2" class="node">
<title>Data_Minig</title>
<polygon fill="none" stroke="black" points="143.66,-196 60.34,-196 60.34,-160 143.66,-160 143.66,-196"></polygon>
<text text-anchor="middle" x="102" y="-173.8" font-family="Times,serif" font-size="14.00">Data_Minig</text>
</g>
<!-- Exploratory_Data_Analysis&#45;&gt;Data_Minig -->
<g id="edge1" class="edge">
<title>Exploratory_Data_Analysis-&gt;Data_Minig</title>
<path fill="none" stroke="black" d="M102,-231.7C102,-223.98 102,-214.71 102,-206.11"></path>
<polygon fill="black" stroke="black" points="105.5,-206.1 102,-196.1 98.5,-206.1 105.5,-206.1"></polygon>
</g>
<!-- Statistical_Analysis -->
<g id="node3" class="node">
<title>Statistical_Analysis</title>
<polygon fill="none" stroke="black" points="165.71,-124 38.29,-124 38.29,-88 165.71,-88 165.71,-124"></polygon>
<text text-anchor="middle" x="102" y="-101.8" font-family="Times,serif" font-size="14.00">Statistical_Analysis</text>
</g>
<!-- Data_Minig&#45;&gt;Statistical_Analysis -->
<g id="edge2" class="edge">
<title>Data_Minig-&gt;Statistical_Analysis</title>
<path fill="none" stroke="black" d="M102,-159.7C102,-151.98 102,-142.71 102,-134.11"></path>
<polygon fill="black" stroke="black" points="105.5,-134.1 102,-124.1 98.5,-134.1 105.5,-134.1"></polygon>
</g>
<!-- Machine_Learning -->
<g id="node4" class="node">
<title>Machine_Learning</title>
<polygon fill="none" stroke="black" points="163.01,-52 40.99,-52 40.99,-16 163.01,-16 163.01,-52"></polygon>
<text text-anchor="middle" x="102" y="-29.8" font-family="Times,serif" font-size="14.00">Machine_Learning</text>
</g>
<!-- Statistical_Analysis&#45;&gt;Machine_Learning -->
<g id="edge3" class="edge">
<title>Statistical_Analysis-&gt;Machine_Learning</title>
<path fill="none" stroke="black" d="M102,-87.7C102,-79.98 102,-70.71 102,-62.11"></path>
<polygon fill="black" stroke="black" points="105.5,-62.1 102,-52.1 98.5,-62.1 105.5,-62.1"></polygon>
</g>
</g>
</svg>
</div>
<p></p></figure><p></p>
</div>
</div>
</div>
<p>Data 분석은 크게 EDA (Exploratory Data Analysis), Statistical Analysis 및 Machine Learning과 같이 3 단계로 수행했다. 각 단계에서 나온 결과가 각 각의 단계에서 일관되게 나오는 metabolites를 선별했다.</p>
<section id="eda-and-data-mining" class="level4">
<h4 class="anchored" data-anchor-id="eda-and-data-mining">EDA and Data Mining</h4>
<p>student t tests, Wilcoxon Man Whiteney tests, <span class="math inline">\(\chi^2\)</span> tests, Fisher Exact Tests, ANOVAs, Kruskal Wallis Tests 및 regression analysis이 수행됐고 visualization을 통해 검정 결과를 재확인하는 작업을 수행했다. 고차원 데이터를 시각화하여 data의 pattern을 관찰하기 위해 KNN, PCA, K means clustering 및 DB Scan을 이용했다.</p>
</section>
<section id="statistical-analysis" class="level4">
<h4 class="anchored" data-anchor-id="statistical-analysis">Statistical Analysis</h4>
<p>multivariable linear regression, logistic regression 및 Cox PH(Proportional Hazards) regression anayses 가 수행됐고 질병과 유의한 metabolites를 선별했다. multiple testing으로 인한 1 종 오류를 범하는 것을 줄이기 위해 permuted p-values를 계산하여 유의성을 한번 더 확인했다.</p>
</section>
<section id="machine-learning" class="level4">
<h4 class="anchored" data-anchor-id="machine-learning">Machine Learning</h4>
<p>Lasso, ridge regression, elastic net, decision tree, random rorests, ada boosting, gradient descent boosting, SVM (support vector machine), partial least square 및 sparse partial least square가 사용됐다. 질병을 가장 잘 예측하는 classifier를 평가하여 최적의 classifier를 선택했다.</p>
</section>
</section>
<section id="results-conclusion" class="level3" data-number="4.4">
<h3 data-number="4.4" class="anchored" data-anchor-id="results-conclusion"><span class="header-section-number">4.4</span> Results &amp; Conclusion</h3>
<p>146개의 관측치와 약 3,000여개의 변수로 구성된 data에서 약 60개 내외의 대사물질이 질병과 5% 유의수준으로 유의한 관계가 있는 것으로 관찰됐고 partial least suare 가 가장 성능이 좋은 것으로 관찰됐다.</p>
</section>
<section id="achivement" class="level3" data-number="4.5">
<h3 data-number="4.5" class="anchored" data-anchor-id="achivement"><span class="header-section-number">4.5</span> Achivement</h3>
<ul>
<li>8개월 동안 연구소에서 파악하지 못한 강력한 교란자를 EDA와 데이터 마이닝을 통해 발견</li>
<li>Columbia University의 Mailman School of Public Health 연례 연구 발표회에서 포스터 발표</li>
<li>연례 연구 경진대회에서 약 100명의 대학원생 중 상위 3명으로 선정되어 상금 $1,000 수상</li>
<li>학과장상 수상</li>
<li>Columbia University Irving Medical Center 신경외과 Job Offer</li>
</ul>
</section>
</section>
</div>
<div id="English" class="tab-pane fade" role="tabpanel" aria-labelledby="English-tab">
<section id="background-1" class="level2" data-number="5">
<h2 data-number="5" class="anchored" data-anchor-id="background-1"><span class="header-section-number">5</span> Background</h2>
<p>Alzheimer Disease (AD) is the most common form of dementia that affects millions of Americans. AD affects memory, thinking and behavior, but its progression is slow, spanning nearly two decades before the symptoms appear. Thus, it is imperative to understand the physiology at the pre-clinical stage. It is estimated that genetic factors contribute nearly 50% to AD. To better understand how genes contribute to the risk of AD by altering cellular milieu, I have examined the metabolome of individuals with the AD-related genotype, APOE. The metabolome represents the products that were generated from the genome and proteome. These biochemical products represent influences of both genetic and environmental factors. The population is Caucasian participants living in longevity village.</p>
</section>
<section id="objective-1" class="level2" data-number="6">
<h2 data-number="6" class="anchored" data-anchor-id="objective-1"><span class="header-section-number">6</span> Objective</h2>
<p>The objective of the Long Life Family Study (LLFS) project was to build an analysis pipeline of identifying significant biomarkers for AD using statistics and machine learning at the multi-stages from the genomic to the metabolomic stage through the transcriptomic and proteomic stage.</p>
</section>
<section id="analysis-pipeline-architecture" class="level2" data-number="7">
<h2 data-number="7" class="anchored" data-anchor-id="analysis-pipeline-architecture"><span class="header-section-number">7</span> Analysis Pipeline Architecture</h2>
<section id="overview-1" class="level3" data-number="7.1">
<h3 data-number="7.1" class="anchored" data-anchor-id="overview-1"><span class="header-section-number">7.1</span> Overview</h3>
<div class="cell" data-layout-align="default">
<div class="cell-output-display">
<div>
<p></p><figure class="figure"><p></p>
<div>
<svg width="672" height="480" viewbox="0.00 0.00 778.83 425.00" xmlns="http://www.w3.org/2000/svg" xlink="http://www.w3.org/1999/xlink" style="; max-width: none; max-height: none">
<g id="graph0" class="graph" transform="scale(1 1) rotate(0) translate(4 421)">
<title>G</title>
<polygon fill="white" stroke="transparent" points="-4,4 -4,-421 774.83,-421 774.83,4 -4,4"></polygon>
<g id="clust4" class="cluster">
<title>cluster3</title>
<polygon fill="none" stroke="black" points="608.08,-8 608.08,-85 770.83,-85 770.83,-8 608.08,-8"></polygon>
<text text-anchor="middle" x="689.46" y="-68.4" font-family="Times,serif" font-size="14.00">Reporting and Conclusion</text>
</g>
<g id="clust2" class="cluster">
<title>cluster1</title>
<polygon fill="none" stroke="black" points="250.54,-170 250.54,-301 424.86,-301 424.86,-170 250.54,-170"></polygon>
<text text-anchor="middle" x="337.7" y="-284.4" font-family="Times,serif" font-size="14.00">Quality Control</text>
</g>
<g id="clust1" class="cluster">
<title>cluster0</title>
<polygon fill="none" stroke="black" points="8,-224 8,-409 230.54,-409 230.54,-224 8,-224"></polygon>
<text text-anchor="middle" x="119.27" y="-392.4" font-family="Times,serif" font-size="14.00">Data Collection</text>
</g>
<g id="clust3" class="cluster">
<title>cluster2</title>
<polygon fill="none" stroke="black" points="444.86,-8 444.86,-247 588.08,-247 588.08,-8 444.86,-8"></polygon>
<text text-anchor="middle" x="516.47" y="-230.4" font-family="Times,serif" font-size="14.00">Analytics</text>
</g>
<!-- Multi_Centered_Blood_Sampling -->
<g id="node1" class="node">
<title>Multi_Centered_Blood_Sampling</title>
<polygon fill="none" stroke="black" points="222.32,-376 16.23,-376 16.23,-340 222.32,-340 222.32,-376"></polygon>
<text text-anchor="middle" x="119.27" y="-353.8" font-family="Times,serif" font-size="14.00">Multi_Centered_Blood_Sampling</text>
</g>
<!-- Mass_Spectrometry -->
<g id="node2" class="node">
<title>Mass_Spectrometry</title>
<polygon fill="none" stroke="black" points="183.25,-322 55.29,-322 55.29,-286 183.25,-286 183.25,-322"></polygon>
<text text-anchor="middle" x="119.27" y="-299.8" font-family="Times,serif" font-size="14.00">Mass_Spectrometry</text>
</g>
<!-- Data_Transfer -->
<g id="node3" class="node">
<title>Data_Transfer</title>
<polygon fill="none" stroke="black" points="167.61,-268 70.93,-268 70.93,-232 167.61,-232 167.61,-268"></polygon>
<text text-anchor="middle" x="119.27" y="-245.8" font-family="Times,serif" font-size="14.00">Data_Transfer</text>
</g>
<!-- Identify_Anomaly_Data -->
<g id="node4" class="node">
<title>Identify_Anomaly_Data</title>
<polygon fill="none" stroke="black" points="414.05,-268 261.35,-268 261.35,-232 414.05,-232 414.05,-268"></polygon>
<text text-anchor="middle" x="337.7" y="-245.8" font-family="Times,serif" font-size="14.00">Identify_Anomaly_Data</text>
</g>
<!-- Data_Transfer&#45;&gt;Identify_Anomaly_Data -->
<g id="edge1" class="edge">
<title>Data_Transfer-&gt;Identify_Anomaly_Data</title>
<path fill="none" stroke="black" d="M167.77,-250C189.1,-250 215.09,-250 240.22,-250"></path>
<polygon fill="black" stroke="black" points="240.55,-253.5 250.55,-250 240.55,-246.5 240.55,-253.5"></polygon>
</g>
<!-- Identify_Missing_Values -->
<g id="node5" class="node">
<title>Identify_Missing_Values</title>
<polygon fill="none" stroke="black" points="417.02,-214 258.39,-214 258.39,-178 417.02,-178 417.02,-214"></polygon>
<text text-anchor="middle" x="337.7" y="-191.8" font-family="Times,serif" font-size="14.00">Identify_Missing_Values</text>
</g>
<!-- EDA -->
<g id="node6" class="node">
<title>EDA</title>
<polygon fill="none" stroke="black" points="543.47,-214 489.47,-214 489.47,-178 543.47,-178 543.47,-214"></polygon>
<text text-anchor="middle" x="516.47" y="-191.8" font-family="Times,serif" font-size="14.00">EDA</text>
</g>
<!-- Identify_Missing_Values&#45;&gt;EDA -->
<g id="edge2" class="edge">
<title>Identify_Missing_Values-&gt;EDA</title>
<path fill="none" stroke="black" d="M417.01,-196C422.8,-196 428.66,-196 434.47,-196"></path>
<polygon fill="black" stroke="black" points="434.87,-199.5 444.87,-196 434.87,-192.5 434.87,-199.5"></polygon>
</g>
<!-- Data_Mining -->
<g id="node7" class="node">
<title>Data_Mining</title>
<polygon fill="none" stroke="black" points="561.63,-160 471.31,-160 471.31,-124 561.63,-124 561.63,-160"></polygon>
<text text-anchor="middle" x="516.47" y="-137.8" font-family="Times,serif" font-size="14.00">Data_Mining</text>
</g>
<!-- Statistical_Analysis -->
<g id="node8" class="node">
<title>Statistical_Analysis</title>
<polygon fill="none" stroke="black" points="580.18,-106 452.76,-106 452.76,-70 580.18,-70 580.18,-106"></polygon>
<text text-anchor="middle" x="516.47" y="-83.8" font-family="Times,serif" font-size="14.00">Statistical_Analysis</text>
</g>
<!-- Machine_Learning -->
<g id="node9" class="node">
<title>Machine_Learning</title>
<polygon fill="none" stroke="black" points="577.48,-52 455.46,-52 455.46,-16 577.48,-16 577.48,-52"></polygon>
<text text-anchor="middle" x="516.47" y="-29.8" font-family="Times,serif" font-size="14.00">Machine_Learning</text>
</g>
<!-- Share_with_Faculty -->
<g id="node10" class="node">
<title>Share_with_Faculty</title>
<polygon fill="none" stroke="black" points="753.71,-52 625.2,-52 625.2,-16 753.71,-16 753.71,-52"></polygon>
<text text-anchor="middle" x="689.46" y="-29.8" font-family="Times,serif" font-size="14.00">Share_with_Faculty</text>
</g>
<!-- Machine_Learning&#45;&gt;Share_with_Faculty -->
<g id="edge3" class="edge">
<title>Machine_Learning-&gt;Share_with_Faculty</title>
<path fill="none" stroke="black" d="M577.75,-34C584.35,-34 591.16,-34 598.02,-34"></path>
<polygon fill="black" stroke="black" points="598.08,-37.5 608.08,-34 598.08,-30.5 598.08,-37.5"></polygon>
</g>
</g>
</svg>
</div>
<p></p></figure><p></p>
</div>
</div>
</div>
<p>Data were obtained by digitization through MS Spectromtetry of blood samples from multiple medical centers in New York, Bonston, Pittsburgh, and Denmark for Caucasians residing in longevity villages. After receiving the data in a csv format through various processes, QC (Quality Control) of the data and data analysis were performed. To better understand data, exploratory data analysis (EDA) and data mining were conducted. Based on the analysis findings on data, the machine learning model to explain the data most was selcted. All findings were shared with biostatisticians, medical doctors, biologists, neurologists and epidemiologists at the neurology department and the Taub Institute for Research on Alzheimer’s Disease and the Aging Brain in the Columbia University Irving Medical Center.</p>
</section>
<section id="data-qc" class="level3" data-number="7.2">
<h3 data-number="7.2" class="anchored" data-anchor-id="data-qc"><span class="header-section-number">7.2</span> Data QC</h3>
<div class="cell" data-fig-height="8" data-layout-align="default">
<div class="cell-output-display">
<div>
<p></p><figure class="figure"><p></p>
<div>
<svg width="672" height="768" viewbox="0.00 0.00 268.00 838.40" xmlns="http://www.w3.org/2000/svg" xlink="http://www.w3.org/1999/xlink" style="; max-width: none; max-height: none">
<g id="graph0" class="graph" transform="scale(1 1) rotate(0) translate(4 834.4)">
<title>G</title>
<polygon fill="white" stroke="transparent" points="-4,4 -4,-834.4 264,-834.4 264,4 -4,4"></polygon>
<g id="clust1" class="cluster">
<title>cluster0</title>
<polygon fill="none" stroke="black" points="8,-236.8 8,-822.4 252,-822.4 252,-236.8 8,-236.8"></polygon>
<text text-anchor="middle" x="130" y="-805.8" font-family="Times,serif" font-size="14.00">Data Quality Control</text>
</g>
<g id="clust2" class="cluster">
<title>cluster1</title>
<polygon fill="none" stroke="black" points="18,-452.8 18,-673.6 242,-673.6 242,-452.8 18,-452.8"></polygon>
<text text-anchor="middle" x="130" y="-657" font-family="Times,serif" font-size="14.00">Missing Value Analysis</text>
</g>
<g id="clust3" class="cluster">
<title>cluster2</title>
<polygon fill="none" stroke="black" points="54,-8 54,-228.8 206,-228.8 206,-8 54,-8"></polygon>
<text text-anchor="middle" x="130" y="-212.2" font-family="Times,serif" font-size="14.00">Data Preprocessing</text>
</g>
<!-- identify_anomaly_data -->
<g id="node1" class="node">
<title>identify_anomaly_data</title>
<polygon fill="none" stroke="black" points="202.57,-789.6 57.43,-789.6 57.43,-753.6 202.57,-753.6 202.57,-789.6"></polygon>
<text text-anchor="middle" x="130" y="-767.4" font-family="Times,serif" font-size="14.00">identify_anomaly_data</text>
</g>
<!-- identify_missing_values -->
<g id="node2" class="node">
<title>identify_missing_values</title>
<polygon fill="none" stroke="black" points="206.38,-717.6 53.62,-717.6 53.62,-681.6 206.38,-681.6 206.38,-717.6"></polygon>
<text text-anchor="middle" x="130" y="-695.4" font-family="Times,serif" font-size="14.00">identify_missing_values</text>
</g>
<!-- identify_anomaly_data&#45;&gt;identify_missing_values -->
<g id="edge3" class="edge">
<title>identify_anomaly_data-&gt;identify_missing_values</title>
<path fill="none" stroke="black" d="M130,-753.3C130,-745.58 130,-736.31 130,-727.71"></path>
<polygon fill="black" stroke="black" points="133.5,-727.7 130,-717.7 126.5,-727.7 133.5,-727.7"></polygon>
</g>
<!-- Missing_Completely_At_Random -->
<g id="node3" class="node">
<title>Missing_Completely_At_Random</title>
<polygon fill="none" stroke="black" points="234.39,-640.8 25.61,-640.8 25.61,-604.8 234.39,-604.8 234.39,-640.8"></polygon>
<text text-anchor="middle" x="130" y="-618.6" font-family="Times,serif" font-size="14.00">Missing_Completely_At_Random</text>
</g>
<!-- identify_missing_values&#45;&gt;Missing_Completely_At_Random -->
<g id="edge4" class="edge">
<title>identify_missing_values-&gt;Missing_Completely_At_Random</title>
<path fill="none" stroke="black" d="M130,-681.25C130,-681.1 130,-680.95 130,-680.8"></path>
<polygon fill="black" stroke="black" points="133.5,-683.6 130,-673.6 126.5,-683.6 133.5,-683.6"></polygon>
</g>
<!-- Missing_At_Random -->
<g id="node4" class="node">
<title>Missing_At_Random</title>
<polygon fill="none" stroke="black" points="198.07,-568.8 61.93,-568.8 61.93,-532.8 198.07,-532.8 198.07,-568.8"></polygon>
<text text-anchor="middle" x="130" y="-546.6" font-family="Times,serif" font-size="14.00">Missing_At_Random</text>
</g>
<!-- Missing_Completely_At_Random&#45;&gt;Missing_At_Random -->
<g id="edge1" class="edge">
<title>Missing_Completely_At_Random-&gt;Missing_At_Random</title>
<path fill="none" stroke="black" d="M130,-604.5C130,-596.78 130,-587.51 130,-578.91"></path>
<polygon fill="black" stroke="black" points="133.5,-578.9 130,-568.9 126.5,-578.9 133.5,-578.9"></polygon>
</g>
<!-- Missing_Not_at_Random -->
<g id="node5" class="node">
<title>Missing_Not_at_Random</title>
<polygon fill="none" stroke="black" points="210.17,-496.8 49.83,-496.8 49.83,-460.8 210.17,-460.8 210.17,-496.8"></polygon>
<text text-anchor="middle" x="130" y="-474.6" font-family="Times,serif" font-size="14.00">Missing_Not_at_Random</text>
</g>
<!-- Missing_At_Random&#45;&gt;Missing_Not_at_Random -->
<g id="edge2" class="edge">
<title>Missing_At_Random-&gt;Missing_Not_at_Random</title>
<path fill="none" stroke="black" d="M130,-532.5C130,-524.78 130,-515.51 130,-506.91"></path>
<polygon fill="black" stroke="black" points="133.5,-506.9 130,-496.9 126.5,-506.9 133.5,-506.9"></polygon>
</g>
<!-- either_imputation_or_omission -->
<g id="node6" class="node">
<title>either_imputation_or_omission</title>
<polygon fill="none" stroke="black" points="225.76,-424.8 34.24,-424.8 34.24,-388.8 225.76,-388.8 225.76,-424.8"></polygon>
<text text-anchor="middle" x="130" y="-402.6" font-family="Times,serif" font-size="14.00">either_imputation_or_omission</text>
</g>
<!-- Missing_Not_at_Random&#45;&gt;either_imputation_or_omission -->
<g id="edge5" class="edge">
<title>Missing_Not_at_Random-&gt;either_imputation_or_omission</title>
<path fill="none" stroke="black" d="M130,-460.5C130,-452.78 130,-443.51 130,-434.91"></path>
<polygon fill="black" stroke="black" points="133.5,-434.9 130,-424.9 126.5,-434.9 133.5,-434.9"></polygon>
</g>
<!-- communication_with_labs -->
<g id="node7" class="node">
<title>communication_with_labs</title>
<polygon fill="none" stroke="black" points="212.03,-352.8 47.97,-352.8 47.97,-316.8 212.03,-316.8 212.03,-352.8"></polygon>
<text text-anchor="middle" x="130" y="-330.6" font-family="Times,serif" font-size="14.00">communication_with_labs</text>
</g>
<!-- either_imputation_or_omission&#45;&gt;communication_with_labs -->
<g id="edge6" class="edge">
<title>either_imputation_or_omission-&gt;communication_with_labs</title>
<path fill="none" stroke="black" d="M130,-388.5C130,-380.78 130,-371.51 130,-362.91"></path>
<polygon fill="black" stroke="black" points="133.5,-362.9 130,-352.9 126.5,-362.9 133.5,-362.9"></polygon>
</g>
<!-- set_data_inclusion_exclusion_criteria -->
<g id="node8" class="node">
<title>set_data_inclusion_exclusion_criteria</title>
<polygon fill="none" stroke="black" points="243.99,-280.8 16.01,-280.8 16.01,-244.8 243.99,-244.8 243.99,-280.8"></polygon>
<text text-anchor="middle" x="130" y="-258.6" font-family="Times,serif" font-size="14.00">set_data_inclusion_exclusion_criteria</text>
</g>
<!-- communication_with_labs&#45;&gt;set_data_inclusion_exclusion_criteria -->
<g id="edge7" class="edge">
<title>communication_with_labs-&gt;set_data_inclusion_exclusion_criteria</title>
<path fill="none" stroke="black" d="M130,-316.5C130,-308.78 130,-299.51 130,-290.91"></path>
<polygon fill="black" stroke="black" points="133.5,-290.9 130,-280.9 126.5,-290.9 133.5,-290.9"></polygon>
</g>
<!-- Data_Transformation -->
<g id="node9" class="node">
<title>Data_Transformation</title>
<polygon fill="none" stroke="black" points="198.01,-196 61.99,-196 61.99,-160 198.01,-160 198.01,-196"></polygon>
<text text-anchor="middle" x="130" y="-173.8" font-family="Times,serif" font-size="14.00">Data_Transformation</text>
</g>
<!-- set_data_inclusion_exclusion_criteria&#45;&gt;Data_Transformation -->
<g id="edge10" class="edge">
<title>set_data_inclusion_exclusion_criteria-&gt;Data_Transformation</title>
<path fill="none" stroke="black" d="M130,-244.64C130,-242.86 130,-240.97 130,-239.02"></path>
<polygon fill="black" stroke="black" points="133.5,-238.8 130,-228.8 126.5,-238.8 133.5,-238.8"></polygon>
</g>
<!-- Log_Transformation -->
<g id="node10" class="node">
<title>Log_Transformation</title>
<polygon fill="none" stroke="black" points="196.14,-124 63.86,-124 63.86,-88 196.14,-88 196.14,-124"></polygon>
<text text-anchor="middle" x="130" y="-101.8" font-family="Times,serif" font-size="14.00">Log_Transformation</text>
</g>
<!-- Data_Transformation&#45;&gt;Log_Transformation -->
<g id="edge8" class="edge">
<title>Data_Transformation-&gt;Log_Transformation</title>
<path fill="none" stroke="black" d="M130,-159.7C130,-151.98 130,-142.71 130,-134.11"></path>
<polygon fill="black" stroke="black" points="133.5,-134.1 130,-124.1 126.5,-134.1 133.5,-134.1"></polygon>
</g>
<!-- Standardization -->
<g id="node11" class="node">
<title>Standardization</title>
<polygon fill="none" stroke="black" points="181.86,-52 78.14,-52 78.14,-16 181.86,-16 181.86,-52"></polygon>
<text text-anchor="middle" x="130" y="-29.8" font-family="Times,serif" font-size="14.00">Standardization</text>
</g>
<!-- Log_Transformation&#45;&gt;Standardization -->
<g id="edge9" class="edge">
<title>Log_Transformation-&gt;Standardization</title>
<path fill="none" stroke="black" d="M130,-87.7C130,-79.98 130,-70.71 130,-62.11"></path>
<polygon fill="black" stroke="black" points="133.5,-62.1 130,-52.1 126.5,-62.1 133.5,-62.1"></polygon>
</g>
</g>
</svg>
</div>
<p></p></figure><p></p>
</div>
</div>
</div>
<p>For data quality control, I communicated with whom generated the data, classified missing values ​​and outliers according to laboratory standards, and labeled them. Based on the results through missing value analysis, I processed the missing values through consultation with the faculty members several times. For the data QC criteria, patients and variables whose ratio of missing values ​​for the sum of the rowwise and columnwise sums exceeded 5% for the sample size were excluded from the analysis. All metabolites data were standardized through log transformation and standardization.</p>
</section>
<section id="methods-1" class="level3" data-number="7.3">
<h3 data-number="7.3" class="anchored" data-anchor-id="methods-1"><span class="header-section-number">7.3</span> Methods</h3>
<div class="cell" data-layout-align="default">
<div class="cell-output-display">
<div>
<p></p><figure class="figure"><p></p>
<div>
<svg width="672" height="480" viewbox="0.00 0.00 212.00 316.80" xmlns="http://www.w3.org/2000/svg" xlink="http://www.w3.org/1999/xlink" style="; max-width: none; max-height: none">
<g id="graph0" class="graph" transform="scale(1 1) rotate(0) translate(4 312.8)">
<title>G</title>
<polygon fill="white" stroke="transparent" points="-4,4 -4,-312.8 208,-312.8 208,4 -4,4"></polygon>
<g id="clust1" class="cluster">
<title>cluster2</title>
<polygon fill="none" stroke="black" points="8,-8 8,-300.8 196,-300.8 196,-8 8,-8"></polygon>
<text text-anchor="middle" x="102" y="-284.2" font-family="Times,serif" font-size="14.00">Data Analytics</text>
</g>
<!-- Exploratory_Data_Analysis -->
<g id="node1" class="node">
<title>Exploratory_Data_Analysis</title>
<polygon fill="none" stroke="black" points="188.3,-268 15.7,-268 15.7,-232 188.3,-232 188.3,-268"></polygon>
<text text-anchor="middle" x="102" y="-245.8" font-family="Times,serif" font-size="14.00">Exploratory_Data_Analysis</text>
</g>
<!-- Data_Minig -->
<g id="node2" class="node">
<title>Data_Minig</title>
<polygon fill="none" stroke="black" points="143.66,-196 60.34,-196 60.34,-160 143.66,-160 143.66,-196"></polygon>
<text text-anchor="middle" x="102" y="-173.8" font-family="Times,serif" font-size="14.00">Data_Minig</text>
</g>
<!-- Exploratory_Data_Analysis&#45;&gt;Data_Minig -->
<g id="edge1" class="edge">
<title>Exploratory_Data_Analysis-&gt;Data_Minig</title>
<path fill="none" stroke="black" d="M102,-231.7C102,-223.98 102,-214.71 102,-206.11"></path>
<polygon fill="black" stroke="black" points="105.5,-206.1 102,-196.1 98.5,-206.1 105.5,-206.1"></polygon>
</g>
<!-- Statistical_Analysis -->
<g id="node3" class="node">
<title>Statistical_Analysis</title>
<polygon fill="none" stroke="black" points="165.71,-124 38.29,-124 38.29,-88 165.71,-88 165.71,-124"></polygon>
<text text-anchor="middle" x="102" y="-101.8" font-family="Times,serif" font-size="14.00">Statistical_Analysis</text>
</g>
<!-- Data_Minig&#45;&gt;Statistical_Analysis -->
<g id="edge2" class="edge">
<title>Data_Minig-&gt;Statistical_Analysis</title>
<path fill="none" stroke="black" d="M102,-159.7C102,-151.98 102,-142.71 102,-134.11"></path>
<polygon fill="black" stroke="black" points="105.5,-134.1 102,-124.1 98.5,-134.1 105.5,-134.1"></polygon>
</g>
<!-- Machine_Learning -->
<g id="node4" class="node">
<title>Machine_Learning</title>
<polygon fill="none" stroke="black" points="163.01,-52 40.99,-52 40.99,-16 163.01,-16 163.01,-52"></polygon>
<text text-anchor="middle" x="102" y="-29.8" font-family="Times,serif" font-size="14.00">Machine_Learning</text>
</g>
<!-- Statistical_Analysis&#45;&gt;Machine_Learning -->
<g id="edge3" class="edge">
<title>Statistical_Analysis-&gt;Machine_Learning</title>
<path fill="none" stroke="black" d="M102,-87.7C102,-79.98 102,-70.71 102,-62.11"></path>
<polygon fill="black" stroke="black" points="105.5,-62.1 102,-52.1 98.5,-62.1 105.5,-62.1"></polygon>
</g>
</g>
</svg>
</div>
<p></p></figure><p></p>
</div>
</div>
</div>
<p>Data analysis was performed in three stages: Exploratory Data Analysis (EDA), Statistical Analysis, and Machine Learning. In each stage, metabolites commonly associated with diseases were selected.</p>
<section id="eda-and-data-mining-1" class="level4">
<h4 class="anchored" data-anchor-id="eda-and-data-mining-1">EDA and Data Mining</h4>
<p>Student t tests, Wilcoxon Man Whiteney tests, <span class="math inline">\(\chi^2\)</span> tests, Fisher Exact Tests, ANOVAs, Kruskal Wallis Tests, and regression testing were performed, and I visualizaed data to reconfirm the test results. To visualize high-dimensional data and observe data patterns, KNN, PCA, K means, Clustering, and DB Scan were used.</p>
</section>
<section id="statistical-analysis-1" class="level4">
<h4 class="anchored" data-anchor-id="statistical-analysis-1">Statistical Analysis</h4>
<p>Multivariable linear regression, logistic regression, and Cox PH (Proportional Hazards) regression analyses were conducted and the metabolites that are signficantly associated with the disease status were selected. In order to reduce the possibility of making a type 1 error due to multiple testing, the significance was checked once more by calculating permuted p-values.</p>
</section>
<section id="machine-learning-1" class="level4">
<h4 class="anchored" data-anchor-id="machine-learning-1">Machine Learning</h4>
<p>Lasso, ridge regression, elastic net, decision tree, random rorests, ada boosting, gradient descent boosting, support vector machine (SVM), partial least square, and sparse partial least square were used. The optimal classifier was selected by evaluating the classifier that best predicted the disease status.</p>
</section>
</section>
<section id="results-conclusion-1" class="level3" data-number="7.4">
<h3 data-number="7.4" class="anchored" data-anchor-id="results-conclusion-1"><span class="header-section-number">7.4</span> Results &amp; Conclusion</h3>
<p>In the data consisting of 146 observations and about 3,000 variables, about 60 metabolites were observed to have a significant relationship with the disease at the 5% significance level, and partial least suare was observed to perform the best.</p>
</section>
</section>
</div>



</div></ul></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>